<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.30">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>15&nbsp; Two-Sample tests – Computational Analysis for Bioscientists</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./one-way-anova-and-kw.html" rel="next">
<link href="./single-linear-regression.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2ee26349945d0c4eadaece612797c059.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-7a3e59189d9f0ecca6f4f741a2329820.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2ee26349945d0c4eadaece612797c059.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-235ce19765b56125a6539188bd174cdb.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-45c5d498b08e9400af60c638f13658fe.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="site_libs/bootstrap/bootstrap-235ce19765b56125a6539188bd174cdb.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="site_libs/kePrint-0.0.1/kePrint.js"></script><link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-sidebar floating quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./statistical-analysis-1.html">Statistical Analysis in R - Part 1</a></li><li class="breadcrumb-item"><a href="./two-sample-tests.html"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Two-Sample tests</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Computational Analysis for Bioscientists</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/3mmaRand/comp4biosci" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome!</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">About this book</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./what-they-forgot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">What they forgot to teach you about computers</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./machine-and-os.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Machines and operating systems</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./file-systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Understanding file systems</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./organising-work.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Organising your work</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./getting-started-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Getting started with data analysis in R</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ideas-about-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Ideas about data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./about-R-and-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">About R and RStudio</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./first-steps-rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">First Steps in RStudio</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./workflow-rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Workflow in RStudio</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./import-to-report.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">From importing to reporting</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./recording-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Recording experiemental data</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./statistical-analysis-1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Statistical Analysis in R - Part 1</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./logic-hypothesis-testing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">The logic of hypothesis testing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./confidence-intervals.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Confidence Intervals</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./what-statistical-model.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">What is a statistical model?</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./single-linear-regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Single linear regression</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./two-sample-tests.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Two-Sample tests</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./one-way-anova-and-kw.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">One-way ANOVA and Kruskal-Wallis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./two-way-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Two-way ANOVA</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./association.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Association: Correlation and Contingency</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./goodness-of-fit.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Goodness of fit</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./working-with-gel-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Working with Gel data in R</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PCR-product-verification.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Agarose gels for PCR product verfication</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./molecular-weight-from-western-blot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Finding the molecular weight of proteins migrated on a western blot</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./in-development.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">In development</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./26C-food-fuel-prac2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title"><em>Pseudomonas putida</em> grown in two soil types at two temperatures</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./keyboard-shortcuts-tips.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Keyboard short cuts and other tips</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">References</span></span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#overview" id="toc-overview" class="nav-link active" data-scroll-target="#overview"><span class="header-section-number">15.1</span> Overview</a>
  <ul class="collapse">
<li><a href="#independent-vs.-paired-samples" id="toc-independent-vs.-paired-samples" class="nav-link" data-scroll-target="#independent-vs.-paired-samples"><span class="header-section-number">15.1.1</span> Independent vs.&nbsp;Paired Samples</a></li>
  <li><a href="#t-tests" id="toc-t-tests" class="nav-link" data-scroll-target="#t-tests"><span class="header-section-number">15.1.2</span> <em>T</em>-tests</a></li>
  <li><a href="#model-assumptions" id="toc-model-assumptions" class="nav-link" data-scroll-target="#model-assumptions"><span class="header-section-number">15.1.3</span> Model assumptions</a></li>
  <li><a href="#reporting" id="toc-reporting" class="nav-link" data-scroll-target="#reporting"><span class="header-section-number">15.1.4</span> Reporting</a></li>
  </ul>
</li>
  <li><a href="#your-turn" id="toc-your-turn" class="nav-link" data-scroll-target="#your-turn"><span class="header-section-number">15.2</span> 🎬 Your turn!</a></li>
  <li>
<a href="#two-independent-samples-parametric" id="toc-two-independent-samples-parametric" class="nav-link" data-scroll-target="#two-independent-samples-parametric"><span class="header-section-number">15.3</span> Two independent samples, parametric</a>
  <ul class="collapse">
<li><a href="#import-and-explore" id="toc-import-and-explore" class="nav-link" data-scroll-target="#import-and-explore"><span class="header-section-number">15.3.1</span> Import and explore</a></li>
  <li><a href="#apply-lm" id="toc-apply-lm" class="nav-link" data-scroll-target="#apply-lm"><span class="header-section-number">15.3.2</span> Apply <code>lm()</code></a></li>
  <li><a href="#check-assumptions" id="toc-check-assumptions" class="nav-link" data-scroll-target="#check-assumptions"><span class="header-section-number">15.3.3</span> Check assumptions</a></li>
  <li><a href="#report" id="toc-report" class="nav-link" data-scroll-target="#report"><span class="header-section-number">15.3.4</span> Report</a></li>
  </ul>
</li>
  <li>
<a href="#two-independent-samples-non-parametric" id="toc-two-independent-samples-non-parametric" class="nav-link" data-scroll-target="#two-independent-samples-non-parametric"><span class="header-section-number">15.4</span> Two independent samples, non-parametric</a>
  <ul class="collapse">
<li><a href="#import-and-explore-1" id="toc-import-and-explore-1" class="nav-link" data-scroll-target="#import-and-explore-1"><span class="header-section-number">15.4.1</span> Import and explore</a></li>
  <li><a href="#apply-wilcox.test" id="toc-apply-wilcox.test" class="nav-link" data-scroll-target="#apply-wilcox.test"><span class="header-section-number">15.4.2</span> Apply <code>wilcox.test()</code></a></li>
  <li><a href="#report-1" id="toc-report-1" class="nav-link" data-scroll-target="#report-1"><span class="header-section-number">15.4.3</span> Report</a></li>
  </ul>
</li>
  <li>
<a href="#two-paired-samples-parametric" id="toc-two-paired-samples-parametric" class="nav-link" data-scroll-target="#two-paired-samples-parametric"><span class="header-section-number">15.5</span> Two paired-samples, parametric</a>
  <ul class="collapse">
<li><a href="#import-and-explore-2" id="toc-import-and-explore-2" class="nav-link" data-scroll-target="#import-and-explore-2"><span class="header-section-number">15.5.1</span> Import and explore</a></li>
  <li><a href="#apply-lm-1" id="toc-apply-lm-1" class="nav-link" data-scroll-target="#apply-lm-1"><span class="header-section-number">15.5.2</span> Apply <code>lm()</code></a></li>
  <li><a href="#check-assumptions-1" id="toc-check-assumptions-1" class="nav-link" data-scroll-target="#check-assumptions-1"><span class="header-section-number">15.5.3</span> Check assumptions</a></li>
  <li><a href="#report-2" id="toc-report-2" class="nav-link" data-scroll-target="#report-2"><span class="header-section-number">15.5.4</span> Report</a></li>
  </ul>
</li>
  <li>
<a href="#two-paired-samples-non-parametric" id="toc-two-paired-samples-non-parametric" class="nav-link" data-scroll-target="#two-paired-samples-non-parametric"><span class="header-section-number">15.6</span> Two paired-samples, non-parametric</a>
  <ul class="collapse">
<li><a href="#pivot-wider" id="toc-pivot-wider" class="nav-link" data-scroll-target="#pivot-wider"><span class="header-section-number">15.6.1</span> Pivot wider</a></li>
  <li><a href="#apply-wilcox.test-1" id="toc-apply-wilcox.test-1" class="nav-link" data-scroll-target="#apply-wilcox.test-1"><span class="header-section-number">15.6.2</span> Apply <code>wilcox.test()</code></a></li>
  <li><a href="#report-3" id="toc-report-3" class="nav-link" data-scroll-target="#report-3"><span class="header-section-number">15.6.3</span> Report</a></li>
  </ul>
</li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">15.7</span> Summary</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/two-sample-tests.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="#"><i class="bi bi-chat-left-dots"></i>Make a suggestion</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./statistical-analysis-1.html">Statistical Analysis in R - Part 1</a></li><li class="breadcrumb-item"><a href="./two-sample-tests.html"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Two-Sample tests</span></a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span id="sec-two-sample-tests" class="quarto-section-identifier"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Two-Sample tests</span></span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Complete
</div>
</div>
<div class="callout-body-container callout-body">
<p>You are reading a live document. This page is compete but suggestions for improvements are welcome. Follow the link to ‘Make a suggestion’ to suggest improvements.</p>
</div>
</div>
<section id="overview" class="level2" data-number="15.1"><h2 data-number="15.1" class="anchored" data-anchor-id="overview">
<span class="header-section-number">15.1</span> Overview</h2>
<p>In the last chapter, we explored single linear regression, a technique used when the explanatory variable is continuous. Now, we shift our focus to cases where the explanatory variable is categorical with two groups. For example, we may want to determine if there is a difference in mass between two subspecies of chaffinch or compare marks in two subjects.</p>
<p>To conduct a two-sample test, we use either <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> or <code><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test()</a></code>, depending on whether the assumptions of <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> are met. General linear models applied with <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> are parametric tests, meaning they rely on the normal distribution’s parameters (mean and standard deviation) to determine statistical significance. The null hypothesis typically concerns the mean or the difference between means. For the <em>p</em>-values to be valid, the assumptions must be satisfied.</p>
<p>If these assumptions are not met, we turn to non-parametric tests, which rely on the ranks of values rather than the actual values themselves. Here, the null hypothesis concerns the mean rank instead of the mean. While non-parametric tests are more flexible and applicable in a wider range of scenarios, they tend to be less powerful, meaning they are less likely to detect a true difference when one exists.</p>
<section id="independent-vs.-paired-samples" class="level3" data-number="15.1.1"><h3 data-number="15.1.1" class="anchored" data-anchor-id="independent-vs.-paired-samples">
<span class="header-section-number">15.1.1</span> Independent vs.&nbsp;Paired Samples</h3>
<p>A crucial consideration when conducting tests is whether the values in one group are independent of those in the other. Non-independence occurs when the two measures are linked in some way—for instance, if they come from the same individual, time, or location.</p>
<p>For example, when evaluating a treatment for high blood pressure, we might measure blood pressure before and after treatment on the same individuals. In this case, the before and after measurements are not independent. If pairs of observations across groups share a common factor that makes them more similar to each other than to other observations, the samples are not independent.</p>
<p>We use different testing approaches for independent and non-independent samples to account for this dependency.</p>
</section><section id="t-tests" class="level3" data-number="15.1.2"><h3 data-number="15.1.2" class="anchored" data-anchor-id="t-tests">
<span class="header-section-number">15.1.2</span> <em>T</em>-tests</h3>
<p>A linear model with one explanatory variable with two groups is also known as a <strong>two-sample <em>t</em>-test</strong> when the samples are independent and as a <strong>paired-samples <em>t</em>-test</strong> when they are not. R does have a <code><a href="https://rdrr.io/r/stats/t.test.html">t.test()</a></code> function which allows you to fit a linear model with just two groups. However, here we teach you to use and interpret the <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> function because it is more generalisable. You can use <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> when you have three or more groups or additional explanatory variables. The output of <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> is also in the same form as many other statistical functions in R. This means what you learn in performing <em>t</em>-tests with <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> will help you learn other methods more easily. However, it is definitely not wrong to use <code><a href="https://rdrr.io/r/stats/t.test.html">t.test()</a></code> rather than <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> for two-group situations - the procedures are identical and the <em>p</em>-values will be the same.</p>
</section><section id="model-assumptions" class="level3" data-number="15.1.3"><h3 data-number="15.1.3" class="anchored" data-anchor-id="model-assumptions">
<span class="header-section-number">15.1.3</span> Model assumptions</h3>
<p>The assumptions of the general linear model are that the residuals are normally distributed and have homogeneity of variance. A residual is the difference between the predicted value and the observed value.</p>
<p>If we have a continuous response and a categorical explanatory variable with two groups, we usually apply the general linear model with <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> and <em>then</em> check the assumptions, however, we can sometimes tell when a non-parametric test would be more appropriate before that:</p>
<ul>
<li><p>Use common sense - the response should be continuous (or nearly continuous, see <a href="ideas_about_data.html#theory-and-practice">Ideas about data: Theory and practice</a>). Consider whether you would expect the response to be continuous.</p></li>
<li><p>We expect decimal places and few repeated values.</p></li>
</ul>
<p>To examine the assumptions after fitting the linear model, we plot the residuals and test them against the normal distribution in the same way as we did for single linear regression.</p>
</section><section id="reporting" class="level3" data-number="15.1.4"><h3 data-number="15.1.4" class="anchored" data-anchor-id="reporting">
<span class="header-section-number">15.1.4</span> Reporting</h3>
<p>In reporting the result of two-sample test we give:</p>
<ol type="1">
<li>
<p>the significance of effect - whether there is there a difference between the groups</p>
<ul>
<li>parametric: whether there is there a difference between the groups means</li>
<li>non-parametric: whether there is there a difference between the group medians</li>
</ul>
</li>
<li><p>the direction of effect - which of the means/medians is greater</p></li>
<li>
<p>the magnitude of effect - how big is the difference between the means/medians</p>
<ul>
<li>parametric: the means and standard errors for each group or the mean difference for paired samples</li>
<li>non-parametric: the medians for each group or the median difference for paired samples</li>
</ul>
</li>
</ol>
<p>Figures should reflect what you have said in the statements. Ideally they should show both the raw data and the statistical model:</p>
<ul>
<li>parametric: means and standard errors</li>
<li>non-parametric: boxplots with medians and interquartile range</li>
</ul>
<p>We will explore all of these ideas with some examples.</p>
</section></section><section id="your-turn" class="level2" data-number="15.2"><h2 data-number="15.2" class="anchored" data-anchor-id="your-turn">
<span class="header-section-number">15.2</span> 🎬 Your turn!</h2>
<p>If you want to code along you will need to start a new <a href="workflow_rstudio.html#rstudio-projects">RStudio project</a>, add a <code>data-raw</code> folder and open a new script. You will also need to load the <strong><code>tidyverse</code></strong> package <span class="citation" data-cites="tidyverse">(<a href="references.html#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span>.</p>
</section><section id="two-independent-samples-parametric" class="level2" data-number="15.3"><h2 data-number="15.3" class="anchored" data-anchor-id="two-independent-samples-parametric">
<span class="header-section-number">15.3</span> Two independent samples, parametric</h2>
<p>A number of <a href="https://en.wikipedia.org/wiki/Common_chaffinch">subspecies of the common chaffinch, <em>Fringilla coelebs</em></a>, have been described based principally on the differences in the pattern and colour of the adult male plumage <span class="citation" data-cites="suárez2009">(<a href="references.html#ref-su%C3%A1rez2009" role="doc-biblioref">Suárez et al. 2009</a>)</span>. Two of groups of these subspecies are:</p>
<ul>
<li>the “coelebs group” (<a href="#fig-coelebs" class="quarto-xref">Figure&nbsp;<span>15.1 (a)</span></a>) that occurs in Europe and Asia</li>
<li>the “canariensis group” (<a href="#fig-canariensis" class="quarto-xref">Figure&nbsp;<span>15.1 (b)</span></a>) that occurs on the Canary Islands.</li>
</ul>
<div id="fig-subspecies" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-subspecies-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-subspecies" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-coelebs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure"><div aria-describedby="fig-coelebs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/512px-Chaffinch_(Fringilla_coelebs).jpg" class="lightbox" data-gallery="fig-subspecies" title="Figure&nbsp;15.1&nbsp;(a): F. c. coelebs"><img src="images/512px-Chaffinch_(Fringilla_coelebs).jpg" class="img-fluid figure-img" data-ref-parent="fig-subspecies"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-coelebs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) <em>F. c. coelebs</em>
</figcaption></figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-subspecies" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-canariensis" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure"><div aria-describedby="fig-canariensis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/512px-Fringilla_coelebs_palmae_-_Los_Tilos.jpg" class="lightbox" data-gallery="fig-subspecies" title="Figure&nbsp;15.1&nbsp;(b): F. c. palmae"><img src="images/512px-Fringilla_coelebs_palmae_-_Los_Tilos.jpg" class="img-fluid figure-img" data-ref-parent="fig-subspecies"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-canariensis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) <em>F. c. palmae</em>
</figcaption></figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-subspecies-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.1: Adult male <em>Fringilla coelebs</em> of the coelebs group on the left (Andreas Trepte, CC BY-SA 2.5 <a href="https://creativecommons.org/licenses/by-sa/2.5" class="uri">https://creativecommons.org/licenses/by-sa/2.5</a>, via Wikimedia Commons) and of the canariensis group on the right (H. Zell, CC BY-SA 3.0 <a href="https://creativecommons.org/licenses/by-sa/3.0" class="uri">https://creativecommons.org/licenses/by-sa/3.0</a>, via Wikimedia Commons).
</figcaption></figure>
</div>
<p>The data in <a href="data-raw/chaff.txt">chaff.txt</a> give the masses of twenty individuals from each subspecies. We want to know if the subspecies differ in mass. These groups are independent - there is no link between values in one group and any value in the other group. In this scenario our null hypothesis, <span class="math inline">\(H_0\)</span>, is that there is no difference between the two subspecies in mass or that subspecies has no effect on mass. This is written as: <span class="math inline">\(H_0: \beta_1 = 0\)</span>.</p>
<section id="import-and-explore" class="level3" data-number="15.3.1"><h3 data-number="15.3.1" class="anchored" data-anchor-id="import-and-explore">
<span class="header-section-number">15.3.1</span> Import and explore</h3>
<p>Import the data:</p>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">chaff</span> <span class="op">&lt;-</span> <span class="fu">read_table</span><span class="op">(</span><span class="st">"data-raw/chaff.txt"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead><tr class="header">
<th data-quarto-table-cell-role="th" style="text-align: left; position: sticky; top: 0; background-color: #FFFFFF;">subspecies</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">mass</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">18.3</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">22.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">22.4</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">18.5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">22.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">19.3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">17.8</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">20.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">22.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">16.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">20.7</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">18.7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">22.6</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">21.5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">21.7</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">19.9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">23.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">17.8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">19.5</td>
</tr>
<tr class="even">
<td style="text-align: left;">coelebs</td>
<td style="text-align: right;">24.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">22.7</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">20.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">25.4</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">20.4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">21.6</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">17.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">26.4</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">20.4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">24.7</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">21.8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">23.4</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">24.4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">21.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">23.4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">20.5</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">21.4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">21.5</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">23.7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">23.4</td>
</tr>
<tr class="even">
<td style="text-align: left;">canariensis</td>
<td style="text-align: right;">21.8</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>These data are in tidy format <span class="citation" data-cites="Wickham2014-nl">(<a href="references.html#ref-Wickham2014-nl" role="doc-biblioref">Wickham 2014</a>)</span> - all the mass values are in one column with another column indicating the subspecies. This means they are well formatted for analysis and plotting.</p>
<p>In the first instance, it is always sensible to create a rough plot of our data. This is to give us an overview and help identify if there are any issues like missing or extreme values. It also gives us idea what we are expecting from the analysis which will make it easier for us to identify if we make some mistake in applying that analysis.</p>
<p>Violin plots (<code>geom_violin()</code>), box plots (<code>geom_boxplot()</code>, see <a href="#fig-chaff-rough" class="quarto-xref">Figure&nbsp;<span>15.2</span></a>) or scatter plots (<code>geom_point()</code>) all make good choices for exploratory plotting and it does not matter which of these you choose.</p>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">chaff</span>,</span>
<span>       <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">subspecies</span>, y <span class="op">=</span> <span class="va">mass</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-chaff-rough" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-chaff-rough-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="two-sample-tests_files/figure-html/fig-chaff-rough-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;15.2: The mass of two subspecies of chaffinch. A boxplot is a useful way to get an overview of the data and helps us identify any issues such as missing or extreme values. It also tells us what to expect from the analysis."><img src="two-sample-tests_files/figure-html/fig-chaff-rough-1.png" class="img-fluid figure-img" width="576"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-chaff-rough-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.2: The mass of two subspecies of chaffinch. A boxplot is a useful way to get an overview of the data and helps us identify any issues such as missing or extreme values. It also tells us what to expect from the analysis.
</figcaption></figure>
</div>
</div>
</div>
<p>R will order the groups alphabetically by default.</p>
<p>The figure suggests that the canariensis group is heavier than the coelebs group.</p>
<p>Summarising the data for each subspecies group is the next sensible step. The most useful summary statistics are the means, standard deviations, sample sizes and standard errors. I recommend the <code>group_by()</code> and <code>summarise()</code> approach:</p>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">chaff_summary</span> <span class="op">&lt;-</span> <span class="va">chaff</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">subspecies</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span>,</span>
<span>            std <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span>,</span>
<span>            n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span>,</span>
<span>            se <span class="op">=</span> <span class="va">std</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We have save the results to <code>chaff_summary</code> so that we can use the means and standard errors in our plot later.</p>
<div class="cell">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">chaff_summary</span></span>
<span><span class="co">## # A tibble: 2 × 5</span></span>
<span><span class="co">##   subspecies   mean   std     n    se</span></span>
<span><span class="co">##   &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;</span></span>
<span><span class="co">## 1 canariensis  22.3  2.15    20 0.481</span></span>
<span><span class="co">## 2 coelebs      20.5  2.14    20 0.478</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="apply-lm" class="level3" data-number="15.3.2"><h3 data-number="15.3.2" class="anchored" data-anchor-id="apply-lm">
<span class="header-section-number">15.3.2</span> Apply <code>lm()</code>
</h3>
<p>We can create a two-sample model like this:</p>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">chaff</span>, <span class="va">mass</span> <span class="op">~</span> <span class="va">subspecies</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And examine the model with:</p>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## lm(formula = mass ~ subspecies, data = chaff)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residuals:</span></span>
<span><span class="co">##     Min      1Q  Median      3Q     Max </span></span>
<span><span class="co">## -5.2750 -1.7000 -0.3775  1.6200  4.1250 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Coefficients:</span></span>
<span><span class="co">##                   Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span><span class="co">## (Intercept)        22.2750     0.4795  46.456   &lt;2e-16 ***</span></span>
<span><span class="co">## subspeciescoelebs  -1.7950     0.6781  -2.647   0.0118 *  </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residual standard error: 2.144 on 38 degrees of freedom</span></span>
<span><span class="co">## Multiple R-squared:  0.1557, Adjusted R-squared:  0.1335 </span></span>
<span><span class="co">## F-statistic: 7.007 on 1 and 38 DF,  p-value: 0.01175</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The Estimates in the Coefficients table give:</p>
<ul>
<li><p><code>(Intercept)</code> known as <span class="math inline">\(\beta_0\)</span>. The mean of the canariensis group (<a href="#fig-two-sample-lm-model" class="quarto-xref">Figure&nbsp;<span>15.3</span></a>). Just as the intercept is the value of the <em>y</em> (the response) when the value of <em>x</em> (the explanatory) is zero in a simple linear regression, this is the value of <code>mass</code> when the <code>subspecies</code> is at its first level. The order of the levels is alphabetical by default.</p></li>
<li><p><code>subspeciescoelebs</code>, known as <span class="math inline">\(\beta_1\)</span>, is what needs to be added to the mean of the canariensis group to get the mean of the coelebs group (<a href="#fig-two-sample-lm-model" class="quarto-xref">Figure&nbsp;<span>15.3</span></a>). Just as the slope is amount of <em>y</em> that needs to be added for each unit of <em>x</em> in a simple linear regression, this is the amount of <code>mass</code> that needs to be added when the <code>subspecies</code> goes from its first level to its second level (<em>i.e.</em>, one unit). The <code>subspeciescoelebs</code> estimate is negative so the the coelebs group mean is lower than the canariensis group mean</p></li>
</ul>
<p>The <em>p</em>-values on each line are tests of whether that coefficient is different from zero. Thus it is:</p>
<p><code>subspeciescoelebs  -1.7950     0.6781  -2.647   0.0118 *</code></p>
<p>that tells us the difference between the means is significant.</p>
<p>The <em>F</em> value and <em>p</em>-value in the last line are a test of whether the model as a whole explains a significant amount of variation in the response variable. For a two-sample test, just like a regression, this is exactly equivalent to the test of the slope against zero and the two <em>p</em>-values will be the same.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-two-sample-lm-model" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-two-sample-lm-model-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="two-sample-tests_files/figure-html/fig-two-sample-lm-model-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;15.3: In a two-sample linear model, the first estimate is the intercept which is the mean of the first group. The second estimate is the ‘slope’ which is what has to added to the intercept to get the second group mean. Note that y axis starts at 15 to create more space for the annotations."><img src="two-sample-tests_files/figure-html/fig-two-sample-lm-model-1.png" class="img-fluid figure-img" width="576"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-two-sample-lm-model-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.3: In a two-sample linear model, the first estimate is the intercept which is the mean of the first group. The second estimate is the ‘slope’ which is what has to added to the intercept to get the second group mean. Note that y axis starts at 15 to create more space for the annotations.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="check-assumptions" class="level3" data-number="15.3.3"><h3 data-number="15.3.3" class="anchored" data-anchor-id="check-assumptions">
<span class="header-section-number">15.3.3</span> Check assumptions</h3>
<p>Check the assumptions: All general linear models assume the “residuals” are normally distributed and have “homogeneity” of variance.</p>
<p>Our first check of these assumptions is to use common sense: mass is a continuous variable and we would expect it to be normally distributed thus we would also expect the residuals to be normally distributed.</p>
<p>We then plot the residuals. The <code><a href="https://rdrr.io/r/graphics/plot.default.html">plot()</a></code> function can be used to plot the residuals against the fitted values (See <a href="#fig-lm-plot1" class="quarto-xref">Figure&nbsp;<span>15.4</span></a>). This is a good way to check for homogeneity of variance.</p>
<div class="cell">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">mod</span>, which <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-lm-plot1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-lm-plot1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="two-sample-tests_files/figure-html/fig-lm-plot1-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;15.4: A plot of the residuals against the fitted values shows the points are distributed similarly in each group. This is a good sign for the assumption of homogeneity of variance."><img src="two-sample-tests_files/figure-html/fig-lm-plot1-1.png" class="img-fluid figure-img" width="576"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lm-plot1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.4: A plot of the residuals against the fitted values shows the points are distributed similarly in each group. This is a good sign for the assumption of homogeneity of variance.
</figcaption></figure>
</div>
</div>
</div>
<p>We can also use a histogram to check for normality (See <a href="#fig-lm-plot2" class="quarto-xref">Figure&nbsp;<span>15.5</span></a>).</p>
<div class="cell">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">residuals</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-lm-plot2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-lm-plot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="two-sample-tests_files/figure-html/fig-lm-plot2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;15.5: A histogram of residuals is symetrical and seems consistent with a normal distribution. This is a good sign for the assumption of normally distributed residuals."><img src="two-sample-tests_files/figure-html/fig-lm-plot2-1.png" class="img-fluid figure-img" width="576"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lm-plot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.5: A histogram of residuals is symetrical and seems consistent with a normal distribution. This is a good sign for the assumption of normally distributed residuals.
</figcaption></figure>
</div>
</div>
</div>
<p>Finally, we can use the Shapiro-Wilk test to test for normality.</p>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">mod</span><span class="op">$</span><span class="va">residuals</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  Shapiro-Wilk normality test</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## data:  mod$residuals</span></span>
<span><span class="co">## W = 0.98046, p-value = 0.7067</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The p-value is greater than 0.05 so this test of the normality assumption is not significant.</p>
<p>Taken together, these results suggest that the assumptions of normality and homogeneity of variance are not violated.</p>
</section><section id="report" class="level3" data-number="15.3.4"><h3 data-number="15.3.4" class="anchored" data-anchor-id="report">
<span class="header-section-number">15.3.4</span> Report</h3>
<p>Canariensis chaffinches (<span class="math inline">\(\bar{x} \pm s.e\)</span>: 22.48 <span class="math inline">\(\pm\)</span> 0.48) were significantly heavier than Coelebs (20.28 <span class="math inline">\(\pm\)</span> 0.48 ) (<em>t</em> = 2.65; <em>d.f.</em> = 38; <em>p</em> = 0.012). See <a href="#fig-chaff" class="quarto-xref">Figure&nbsp;<span>15.6</span></a>.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="va">chaff</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">subspecies</span>, y <span class="op">=</span> <span class="va">mass</span><span class="op">)</span>,</span>
<span>             position <span class="op">=</span> <span class="fu">position_jitter</span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.1</span>, height <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>             colour <span class="op">=</span> <span class="st">"gray50"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_errorbar</span><span class="op">(</span>data <span class="op">=</span> <span class="va">chaff_summary</span>, </span>
<span>                <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">subspecies</span>, ymin <span class="op">=</span> <span class="va">mean</span> <span class="op">-</span> <span class="va">se</span>, ymax <span class="op">=</span> <span class="va">mean</span> <span class="op">+</span> <span class="va">se</span><span class="op">)</span>,</span>
<span>                width <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_errorbar</span><span class="op">(</span>data <span class="op">=</span> <span class="va">chaff_summary</span>, </span>
<span>                <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">subspecies</span>, ymin <span class="op">=</span> <span class="va">mean</span>, ymax <span class="op">=</span> <span class="va">mean</span><span class="op">)</span>,</span>
<span>                width <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Mass (g)"</span>, </span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">30</span><span class="op">)</span>, </span>
<span>                     expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Subspecies"</span>, </span>
<span>                   labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Canariensis"</span>, <span class="st">"Coelebs"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"segment"</span>, x <span class="op">=</span> <span class="fl">1</span>, xend <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           y <span class="op">=</span> <span class="fl">28</span>, yend <span class="op">=</span> <span class="fl">28</span>,</span>
<span>           colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">1.5</span>,  y <span class="op">=</span> <span class="fl">29</span>, </span>
<span>           label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu">italic</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">~</span><span class="st">"= 0.012"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="fig-chaff" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-chaff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="two-sample-tests_files/figure-html/unnamed-chunk-9-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;15.6: Canariensis chaffinches are heavier than Coelebs chaffinches. The mean mass of 20 randomly sampled males from each subspecies was determined. Error bars are \pm 1 standard error. Canariensis chaffinches were significantly heavier than Coelebs (t = 2.65; d.f. = 38; p = 0.012). Data analysis was conducted in R [@R-core] with tidyverse packages [@tidyverse]."><img src="two-sample-tests_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="576"></a></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-chaff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.6: <strong>Canariensis chaffinches are heavier than Coelebs chaffinches</strong>. The mean mass of 20 randomly sampled males from each subspecies was determined. Error bars are <span class="math inline">\(\pm\)</span> 1 standard error. Canariensis chaffinches were significantly heavier than Coelebs (<em>t</em> = 2.65; <em>d.f.</em> = 38; <em>p</em> = 0.012). Data analysis was conducted in R <span class="citation" data-cites="R-core">(<a href="references.html#ref-R-core" role="doc-biblioref">R Core Team 2025</a>)</span> with tidyverse packages <span class="citation" data-cites="tidyverse">(<a href="references.html#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span>.
</figcaption></figure>
</div>
</section></section><section id="two-independent-samples-non-parametric" class="level2" data-number="15.4"><h2 data-number="15.4" class="anchored" data-anchor-id="two-independent-samples-non-parametric">
<span class="header-section-number">15.4</span> Two independent samples, non-parametric</h2>
<p>The non-parametric equivalent of the linear model with two independent samples is the “Wilcoxon rank sum test” <span class="citation" data-cites="wilcoxon1945">(<a href="references.html#ref-wilcoxon1945" role="doc-biblioref">Wilcoxon 1945</a>)</span>. It is commonly also known as the Mann-Whitney or Wilcoxon–Mann–Whitney.</p>
<p>The general question you have about your data - are these two groups different - is the same, but one of more of the following is true:</p>
<ul>
<li>the response variable is not continuous</li>
<li>the residuals are not normally distributed</li>
<li>the sample size is too small to tell if they are normally distributed.</li>
<li>the variance is not homogeneous</li>
</ul>
<p>The test is a applied in R with the <code><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test()</a></code> function.</p>
<p>The data in <a href="data-raw/arabidopsis.txt">arabidopsis.txt</a> give the number of leaves on eight wildtype and eight mutant <em>Arabidopsis thaliana</em> plants. We want to know if the two types of plants have differing numbers of leaves. These are counts, so they are not continuous and the sample sizes are quite small. A non-parametric test is a safer option. In this scenario our null hypothesis, <span class="math inline">\(H_0\)</span>, is that there is no difference between the two types of plant in the number of leaves.</p>
<section id="import-and-explore-1" class="level3" data-number="15.4.1"><h3 data-number="15.4.1" class="anchored" data-anchor-id="import-and-explore-1">
<span class="header-section-number">15.4.1</span> Import and explore</h3>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">arabidopsis</span> <span class="op">&lt;-</span> <span class="fu">read_table</span><span class="op">(</span><span class="st">"data-raw/arabidopsis.txt"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>These data are in tidy format <span class="citation" data-cites="Wickham2014-nl">(<a href="references.html#ref-Wickham2014-nl" role="doc-biblioref">Wickham 2014</a>)</span> - the numbers of leaves are in one column with another column indicating whether the observation comes from a wildtype or mutant <em>Arabidopsis</em>. This means they are well formatted for analysis and plotting.</p>
<p>Create a quick plot of the data:</p>
<div class="cell">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">arabidopsis</span>, </span>
<span>       <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">type</span>, y <span class="op">=</span> <span class="va">leaves</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-arabid-rough" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-arabid-rough-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="two-sample-tests_files/figure-html/fig-arabid-rough-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure&nbsp;15.7: The number of leaves on mutant and wildtype plants. A boxplot is a useful way to get an overview of the data and helps us identify any issues such as missing or extreme values. It also tells us what to expect from the analysis."><img src="two-sample-tests_files/figure-html/fig-arabid-rough-1.png" class="img-fluid figure-img" width="576"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-arabid-rough-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.7: The number of leaves on mutant and wildtype plants. A boxplot is a useful way to get an overview of the data and helps us identify any issues such as missing or extreme values. It also tells us what to expect from the analysis.
</figcaption></figure>
</div>
</div>
</div>
<p>Our rough plot shows that the mutant plants have fewer leaves than the wildtype plants.</p>
<p>Summarising the data using the median and interquartile range is more aligned to the type of data and the type of analysis than using means and standard deviations:</p>
<div class="cell">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">arabidopsis_summary</span> <span class="op">&lt;-</span> <span class="va">arabidopsis</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">type</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>median <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">leaves</span><span class="op">)</span>,</span>
<span>            interquartile  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/IQR.html">IQR</a></span><span class="op">(</span><span class="va">leaves</span><span class="op">)</span>,</span>
<span>            n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">leaves</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>View the results:</p>
<div class="cell">
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">arabidopsis_summary</span></span>
<span><span class="co">## # A tibble: 2 × 4</span></span>
<span><span class="co">##   type   median interquartile     n</span></span>
<span><span class="co">##   &lt;chr&gt;   &lt;dbl&gt;         &lt;dbl&gt; &lt;int&gt;</span></span>
<span><span class="co">## 1 mutant    5             2.5     8</span></span>
<span><span class="co">## 2 wild      8.5           1.5     8</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="apply-wilcox.test" class="level3" data-number="15.4.2"><h3 data-number="15.4.2" class="anchored" data-anchor-id="apply-wilcox.test">
<span class="header-section-number">15.4.2</span> Apply <code>wilcox.test()</code>
</h3>
<p>We pass the dataframe and variables to <code><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test()</a></code> in the same way as we did for <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code>. We give the data argument and a “formula” which says <code>leaves ~ type</code> meaning “explain leaves by type”.</p>
<div class="cell">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">arabidopsis</span>, <span class="va">leaves</span> <span class="op">~</span> <span class="va">type</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## data:  leaves by type</span></span>
<span><span class="co">## W = 5, p-value = 0.005051</span></span>
<span><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The warning message “Warning: cannot compute exact p-value with ties” is not something to worry about too much. It is a warning rather than an indication that your results are incorrect. It means the <em>p</em> -value is based on an approximation rather than being exact because there are ties (some values are the same).</p>
<p>The result of the test is given on this line: <code>W = 5, p-value = 0.005051</code>. <code>W</code> is the test statistic. The <em>p</em>-value is less than 0.05 meaning there is a significant difference in the number of leaves on wildtype and mutant plants.</p>
</section><section id="report-1" class="level3" data-number="15.4.3"><h3 data-number="15.4.3" class="anchored" data-anchor-id="report-1">
<span class="header-section-number">15.4.3</span> Report</h3>
<p>There are significantly more leaves on wildtype (median = 8.5) than mutant (median = 5) plants (Wilcoxon rank sum test: <em>W</em> = 5, <span class="math inline">\(n_1\)</span> = 8, <span class="math inline">\(n_2\)</span> = 8, <em>p</em> = 0.005). See <a href="#fig-arabid" class="quarto-xref">Figure&nbsp;<span>15.8</span></a>.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">arabidopsis</span>, </span>
<span>       <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">type</span>, y <span class="op">=</span> <span class="va">leaves</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Number of leaves"</span>, </span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">12</span><span class="op">)</span>, </span>
<span>                     expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">""</span>, </span>
<span>                   labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mutatnt"</span>, <span class="st">"Wildtype"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"segment"</span>, x <span class="op">=</span> <span class="fl">1</span>, xend <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           y <span class="op">=</span> <span class="fl">10.5</span>, yend <span class="op">=</span> <span class="fl">10.5</span>,</span>
<span>           colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">1.5</span>,  y <span class="op">=</span> <span class="fl">11</span>, </span>
<span>           label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu">italic</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">~</span><span class="st">"= 0.005"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="fig-arabid" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-arabid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="two-sample-tests_files/figure-html/unnamed-chunk-14-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Figure&nbsp;15.8: Mutant Arabidopsis thaliana have fewer leaves. There are significantly more leaves on wildtype than mutant plants (Wilcoxon rank sum test: W = 5, n_1 = 8, n_2 = 8, p = 0.005). The heavy lines indicate the median of leaves, boxes indicate the interquartile range and whiskers the range. Data analysis was conducted in R [@R-core] with tidyverse packages [@tidyverse]."><img src="two-sample-tests_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="576"></a></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-arabid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.8: <strong>Mutant <em>Arabidopsis thaliana</em> have fewer leaves</strong>. There are significantly more leaves on wildtype than mutant plants (Wilcoxon rank sum test: <em>W</em> = 5, <span class="math inline">\(n_1\)</span> = 8, <span class="math inline">\(n_2\)</span> = 8, <em>p</em> = 0.005). The heavy lines indicate the median of leaves, boxes indicate the interquartile range and whiskers the range. Data analysis was conducted in R <span class="citation" data-cites="R-core">(<a href="references.html#ref-R-core" role="doc-biblioref">R Core Team 2025</a>)</span> with tidyverse packages <span class="citation" data-cites="tidyverse">(<a href="references.html#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span>.
</figcaption></figure>
</div>
</section></section><section id="two-paired-samples-parametric" class="level2" data-number="15.5"><h2 data-number="15.5" class="anchored" data-anchor-id="two-paired-samples-parametric">
<span class="header-section-number">15.5</span> Two paired-samples, parametric</h2>
<p>The data in <a href="data-raw/marks.csv">marks.csv</a> give the marks for ten students in two subjects: Data Analysis and Biology. These data are paired because we have two marks from one student so that a mark in one group has a closer relationship with one of the marks in the other group than with any of the other values. We want to know if students do equally well in both subjects. In this scenario our null hypothesis, <span class="math inline">\(H_0\)</span>, is that there is no difference between the Data Analysis and Biology marks for a student.</p>
<section id="import-and-explore-2" class="level3" data-number="15.5.1"><h3 data-number="15.5.1" class="anchored" data-anchor-id="import-and-explore-2">
<span class="header-section-number">15.5.1</span> Import and explore</h3>
<p>Import the data:</p>
<div class="cell">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">marks</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data-raw/marks.csv"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Since these data are paired, it makes sense to highlight how the marks differ for each student. One way of doing that is to draw a line linking their marks in each subject. This is known as a spaghetti plot. We can use two geoms: <code>geom_point()</code> and <code>geom_line()</code>. To join a student’s marks, we need to set the <code>group</code> aesthetic to <code>student</code>.<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">marks</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">subject</span>, y <span class="op">=</span> <span class="va">mark</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>group <span class="op">=</span> <span class="va">student</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="two-sample-tests_files/figure-html/unnamed-chunk-16-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10"><img src="two-sample-tests_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="576"></a></p>
</figure>
</div>
</div>
</div>
<p>Summarise the data so that we can use the means in plots later:</p>
<div class="cell">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">marks_summary</span> <span class="op">&lt;-</span> <span class="va">marks</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">subject</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">mark</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>A paired test requires us to into account the variation between students.</p>
</section><section id="apply-lm-1" class="level3" data-number="15.5.2"><h3 data-number="15.5.2" class="anchored" data-anchor-id="apply-lm-1">
<span class="header-section-number">15.5.2</span> Apply <code>lm()</code>
</h3>
<p>We can create a paired-sample model with the <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> function<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> like this:</p>
<div class="cell">
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mark</span> <span class="op">~</span> <span class="va">subject</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">student</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">marks</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>
<code>mark</code> is the dependent variable (response).</li>
<li>
<code>subject</code> is the independent variable (explanatory factor).</li>
<li>
<code>factor(student)</code> accounts for the pairing by treating student as another explanatory variable. We have used factor because the values in students are the numbers 1 to 10 and we want <code>student</code> to be treated as a category not a number</li>
</ul>
<div class="cell">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## lm(formula = mark ~ subject + factor(student), data = marks)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residuals:</span></span>
<span><span class="co">##    Min     1Q Median     3Q    Max </span></span>
<span><span class="co">##   -6.5   -3.5    0.0    3.5    6.5 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Coefficients:</span></span>
<span><span class="co">##                     Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span><span class="co">## (Intercept)           89.500      4.697  19.055 1.39e-08 ***</span></span>
<span><span class="co">## subjectDataAnalysis    7.000      2.832   2.471 0.035486 *  </span></span>
<span><span class="co">## factor(student)2     -39.500      6.333  -6.237 0.000152 ***</span></span>
<span><span class="co">## factor(student)3     -26.500      6.333  -4.184 0.002361 ** </span></span>
<span><span class="co">## factor(student)4     -25.500      6.333  -4.026 0.002990 ** </span></span>
<span><span class="co">## factor(student)5     -16.000      6.333  -2.526 0.032431 *  </span></span>
<span><span class="co">## factor(student)6     -54.000      6.333  -8.526 1.33e-05 ***</span></span>
<span><span class="co">## factor(student)7      -9.000      6.333  -1.421 0.189010    </span></span>
<span><span class="co">## factor(student)8     -27.000      6.333  -4.263 0.002101 ** </span></span>
<span><span class="co">## factor(student)9     -44.000      6.333  -6.947 6.70e-05 ***</span></span>
<span><span class="co">## factor(student)10     -1.500      6.333  -0.237 0.818082    </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residual standard error: 6.333 on 9 degrees of freedom</span></span>
<span><span class="co">## Multiple R-squared:  0.9441, Adjusted R-squared:  0.8821 </span></span>
<span><span class="co">## F-statistic: 15.21 on 10 and 9 DF,  p-value: 0.0001814</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The coefficient for <code>(Intercept)</code> gives the mean Biology mark and that for <code>subjectDataAnalysis</code> is amount that the Data Analysis mark are above Biology marks in general. The <em>p</em>-value tests whether this difference is significantly different from zero. The rest of the output considers how students differ. You can ignore this here.</p>
<p>If you find this a bit overwhelming to read you can use the <code><a href="https://rdrr.io/r/stats/anova.html">anova()</a></code> function on the model object to get a simpler output:</p>
<div class="cell">
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: mark</span></span>
<span><span class="co">##                 Df Sum Sq Mean Sq F value   Pr(&gt;F)    </span></span>
<span><span class="co">## subject          1  245.0  245.00   6.108 0.035486 *  </span></span>
<span><span class="co">## factor(student)  9 5856.2  650.69  16.222 0.000153 ***</span></span>
<span><span class="co">## Residuals        9  361.0   40.11                     </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>You will notice that the <em>p</em>-value for subject is the same. The test statistic, <span class="math inline">\(F\)</span> has a value of 6.11 and degrees of freedom of 1 and 9. This would be written as $F = $ 6.11 ; $d.f.= $ 1, 9; $p = $ 0.035 .</p>
</section><section id="check-assumptions-1" class="level3" data-number="15.5.3"><h3 data-number="15.5.3" class="anchored" data-anchor-id="check-assumptions-1">
<span class="header-section-number">15.5.3</span> Check assumptions</h3>
<p>We might expect marks to be normally distributed. However, this is a very small sample, and choosing a non-parametric test instead would be reasonable. However, we will continue with this example to demonstrate how to interpret and report on the result of a parametric paired-samples test (paired-samples <em>t</em>-test).</p>
<p>A plot the residuals against the fitted values (<code>plot(mod, which = 1)</code>) is not useful for a paired test. The normality of the residuals should be checked.</p>
<div class="cell">
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">residuals</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="two-sample-tests_files/figure-html/unnamed-chunk-21-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11"><img src="two-sample-tests_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="576"></a></p>
</figure>
</div>
</div>
</div>
<p>We only have 10 values, so the distribution is never going to look smooth. We can’t draw strong conclusions from this, but we do at least have a peak at 0. Similarly, a normality test is likely to be non-significant because of the small sample size, meaning the test is not very powerful. This means a non-significant result is not strong evidence of the residuals following a normal distribution:</p>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">mod</span><span class="op">$</span><span class="va">residuals</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  Shapiro-Wilk normality test</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## data:  mod$residuals</span></span>
<span><span class="co">## W = 0.92894, p-value = 0.1473</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="report-2" class="level3" data-number="15.5.4"><h3 data-number="15.5.4" class="anchored" data-anchor-id="report-2">
<span class="header-section-number">15.5.4</span> Report</h3>
<p>Individual students score significantly higher in Data Analysis than in Biology (<em>t</em> = 2.47; <em>d.f.</em> = 9; <em>p</em> = 0.0355) with an average difference of 7%. See <a href="#fig-marks" class="quarto-xref">Figure&nbsp;<span>15.9</span></a></p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">marks</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">subject</span>, y <span class="op">=</span> <span class="va">mark</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>pch <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>group <span class="op">=</span> <span class="va">student</span><span class="op">)</span>, linetype <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="va">marks_summary</span>,</span>
<span>             <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">subject</span>, y <span class="op">=</span> <span class="va">mean</span><span class="op">)</span>,</span>
<span>             size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Mark"</span>,</span>
<span>                     expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">110</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"segment"</span>, x <span class="op">=</span> <span class="fl">1</span>, xend <span class="op">=</span> <span class="fl">2</span>,</span>
<span>           y <span class="op">=</span> <span class="fl">105</span>, yend <span class="op">=</span> <span class="fl">105</span>,</span>
<span>           colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">1.5</span>,  y <span class="op">=</span> <span class="fl">108</span>,</span>
<span>           label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu">italic</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">~</span><span class="st">"= 0.0355"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="fig-marks" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-marks-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="two-sample-tests_files/figure-html/unnamed-chunk-23-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="Figure&nbsp;15.9: Students score higher in Data Analysis than in Biology. Open circles indicate an individual student’s marks in each subject with dashed lines joining their marks in each subject. The filled circles indicate the mean mark for each subject. Individual students score significantly higher in Data Analysis than in Biology (t = 2.47; d.f. = 9; p = 0.0355) with an average difference of 7%. Data analysis was conducted in R [@R-core] with tidyverse packages [@tidyverse]."><img src="two-sample-tests_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="576"></a></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-marks-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca"><strong>Figure</strong>&nbsp;15.9: <strong>Students score higher in Data Analysis than in Biology</strong>. Open circles indicate an individual student’s marks in each subject with dashed lines joining their marks in each subject. The filled circles indicate the mean mark for each subject. Individual students score significantly higher in Data Analysis than in Biology (<em>t</em> = 2.47; <em>d.f.</em> = 9; <em>p</em> = 0.0355) with an average difference of 7%. Data analysis was conducted in R <span class="citation" data-cites="R-core">(<a href="references.html#ref-R-core" role="doc-biblioref">R Core Team 2025</a>)</span> with tidyverse packages <span class="citation" data-cites="tidyverse">(<a href="references.html#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span>.
</figcaption></figure>
</div>
</section></section><section id="two-paired-samples-non-parametric" class="level2" data-number="15.6"><h2 data-number="15.6" class="anchored" data-anchor-id="two-paired-samples-non-parametric">
<span class="header-section-number">15.6</span> Two paired-samples, non-parametric</h2>
<p>We have the marks for just 10 students. This sample is too small for us to judge whether the marks are normally distributed. We will use a non-parametric test instead. The “Wilcoxon signed-rank” test is the non-parametric equivalent of the paired-samples <em>t</em>-test. This is often referred to as the paired-sample Wilcoxon test, or just the Wilcoxon test.</p>
<p>The test is also applied in R with the <code><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test()</a></code> function but we add the <code>paired = TRUE</code> argument. We also have to give the two datasets rather than using the “formula method” of <code>mark ~ subject</code>. This means it is useful to pivot the data to “wide” format.</p>
<section id="pivot-wider" class="level3" data-number="15.6.1"><h3 data-number="15.6.1" class="anchored" data-anchor-id="pivot-wider">
<span class="header-section-number">15.6.1</span> Pivot wider</h3>
<p>Create a new dataframe marks_wide from marks:</p>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">marks_wide</span> <span class="op">&lt;-</span> <span class="va">marks</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>values_from <span class="op">=</span> <span class="va">mark</span>, </span>
<span>              names_from <span class="op">=</span> <span class="va">subject</span>, </span>
<span>              id_cols <span class="op">=</span> <span class="va">student</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>
<code>values_from = mark</code>: Uses the <code>mark</code> column as values in the wide format.</li>
<li>
<code>names_from = subject</code>: Creates new columns based on unique values in <code>subject</code>
</li>
<li>
<code>id_cols = student</code>: Keeps <code>student</code> as the identifier (<em>i.e.</em>, each row represents a student).</li>
</ul></section><section id="apply-wilcox.test-1" class="level3" data-number="15.6.2"><h3 data-number="15.6.2" class="anchored" data-anchor-id="apply-wilcox.test-1">
<span class="header-section-number">15.6.2</span> Apply <code>wilcox.test()</code>
</h3>
<p>To apply a paired test with <code><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test()</a></code> we need to use the wide format data (untidy) and add the <code>paired = TRUE</code> argument:</p>
<div class="cell">
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test</a></span><span class="op">(</span><span class="va">marks_wide</span><span class="op">$</span><span class="va">Biology</span>, <span class="va">marks_wide</span><span class="op">$</span><span class="va">DataAnalysis</span>, paired <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  Wilcoxon signed rank test with continuity correction</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## data:  marks_wide$Biology and marks_wide$DataAnalysis</span></span>
<span><span class="co">## V = 6.5, p-value = 0.03641</span></span>
<span><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="report-3" class="level3" data-number="15.6.3"><h3 data-number="15.6.3" class="anchored" data-anchor-id="report-3">
<span class="header-section-number">15.6.3</span> Report</h3>
<p>Individual students score significantly higher in Data Analysis than in Biology (Wilcoxon signed rank test: <em>V</em> = 6.5; <span class="math inline">\(n\)</span> = 10; <em>p</em> = 0.036).</p>
</section></section><section id="summary" class="level2" data-number="15.7"><h2 data-number="15.7" class="anchored" data-anchor-id="summary">
<span class="header-section-number">15.7</span> Summary</h2>
<ol type="1">
<li><p>A linear model with one explanatory variable with two groups and one continuous response is “a two-sample test”.</p></li>
<li><p>If pairs of observations in the groups have something in common that make them more similar to each other, than to other observations, then those observations are not independent. A <strong>paired-samples test</strong> is used when the observations are not independent.</p></li>
<li><p>A linear model with one explanatory variable with two groups and one continuous response is also known as a <strong>two-sample <em>t</em>-test</strong> when the samples are independent and as a <strong>paired-samples <em>t</em>-test</strong> when they are not</p></li>
<li><p>We can use <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> to do two-sample and paired sample tests. We can also use <code><a href="https://rdrr.io/r/stats/t.test.html">t.test()</a></code> for these but using <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> helps us understand tests with more groups and/or more variables where we will have to use <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code>. The output of <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> is also more typical of the output of statistical functions in R.</p></li>
<li><p>We estimate the <strong>coefficients</strong> (also called the <strong>parameters</strong>) of the model. For a two-sample test these are the mean of the first group, <span class="math inline">\(\beta_0\)</span> (which might also be called the intercept) and the difference between the means of the first and second groups, <span class="math inline">\(\beta_1\)</span> (which might also be called the slope). For a paired-sample test there is just one parameter, the mean difference between pairs of values, <span class="math inline">\(\beta_0\)</span> (which might also be called the intercept). We test whether the parameters differ significantly from zero</p></li>
<li><p>We can use <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> to a linear regression.</p></li>
<li><p>In the output of <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code> the coefficients are listed in a table in the Estimates column. The <em>p</em>-value for each coefficient is in the test of whether it differs from zero. At the bottom of the output there is a test of the model <em>overall</em>. In this case, this is exactly the same as the test of the <span class="math inline">\(\beta_1\)</span> and the p-values are identical. The R-squared value is the proportion of the variance in the response variable that is explained by the model.</p></li>
<li><p>The assumptions of the general linear model are that the residuals are normally distributed and have homogeneity of variance. A residual is the difference between the predicted value and the observed value.</p></li>
<li><p>We examine a histogram of the residuals and use the Shapiro-Wilk normality test to check the normality assumption. We check the variance of the residuals is the same for all fitted values with a residuals vs fitted plot.</p></li>
<li><p>If the assumptions are not met, we can use alternatives known as non-parametric tests. These are applied with <code><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test()</a></code> in R.</p></li>
<li><p>When reporting the results of a test we give the significance, direction and size of the effect. Our figures and the values we give should reflect the type of test we have used. We use means and standard errors for parametric tests and medians and interquartile ranges for non-parametric tests. We also give the test statistic, the degrees of freedom (parametric) or sample size (non-parametric) and the p-value. We annotate our figures with the p-value, making clear which comparison it applies to.</p></li>
</ol>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-R-core" class="csl-entry" role="listitem">
R Core Team. 2025. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-suárez2009" class="csl-entry" role="listitem">
Suárez, Nicolás M., Eva Betancor, Tilman E. Klassert, Teresa Almeida, Mariano Hernández, and José J. Pestano. 2009. <span>“Phylogeography and Genetic Structure of the Canarian Common Chaffinch (Fringilla Coelebs) Inferred with mtDNA and Microsatellite Loci.”</span> <em>Molecular Phylogenetics and Evolution</em> 53 (2): 556–64. <a href="https://doi.org/10.1016/j.ympev.2009.07.018">https://doi.org/10.1016/j.ympev.2009.07.018</a>.
</div>
<div id="ref-Wickham2014-nl" class="csl-entry" role="listitem">
Wickham, Hadley. 2014. <span>“Tidy Data.”</span> <em>Journal of Statistical Software, Articles</em> 59 (10): 1–23. <a href="https://vita.had.co.nz/papers/tidy-data.pdf">https://vita.had.co.nz/papers/tidy-data.pdf</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span class="nocase">tidyverse</span>.”</span> <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-wilcoxon1945" class="csl-entry" role="listitem">
Wilcoxon, Frank. 1945. <span>“Individual Comparisons by Ranking Methods.”</span> <em>Biometrics Bulletin</em> 1 (6): 80–83. <a href="https://doi.org/10.2307/3001968">https://doi.org/10.2307/3001968</a>.
</div>
</div>
</section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><hr>
<ol>
<li id="fn1"><p>You might like to try removing <code>aes(group = student)</code> to see what ggplot does when the lines are <em>not</em> grouped by student.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>This is not the only way to apply a paired test. When there are only two groups and no other explanatory variables, we can use <code>t.test(data = marks, mark ~ subject, paired = TRUE)</code>. A more general method that works when you have two or more non-independent values (e.g., more than two subjects) or additional explanatory variables is to create a “linear mixed model” with <code>lmer()</code>.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./single-linear-regression.html" class="pagination-link" aria-label="Single linear regression">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Single linear regression</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./one-way-anova-and-kw.html" class="pagination-link" aria-label="One-way ANOVA and Kruskal-Wallis">
        <span class="nav-page-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">One-way ANOVA and Kruskal-Wallis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb28" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Two-Sample tests {#sec-two-sample-tests}</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: "asis"</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"_common.R"</span>)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="fu">status</span>(<span class="st">"complete"</span>)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overview</span></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>In the last chapter, we explored single linear regression, a technique</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>used when the explanatory variable is continuous. Now, we shift our</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>focus to cases where the explanatory variable is categorical with two</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>groups. For example, we may want to determine if there is a difference</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>in mass between two subspecies of chaffinch or compare marks in two</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>subjects.</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>To conduct a two-sample test, we use either <span class="in">`lm()`</span> or <span class="in">`wilcox.test()`</span>,</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>depending on whether the assumptions of <span class="in">`lm()`</span> are met. General linear</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>models applied with <span class="in">`lm()`</span> are parametric tests, meaning they rely on</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>the normal distribution's parameters (mean and standard deviation) to</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>determine statistical significance. The null hypothesis typically</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>concerns the mean or the difference between means. For the *p*-values to</span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>be valid, the assumptions must be satisfied.</span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>If these assumptions are not met, we turn to non-parametric tests, which</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>rely on the ranks of values rather than the actual values themselves.</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>Here, the null hypothesis concerns the mean rank instead of the mean.</span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>While non-parametric tests are more flexible and applicable in a wider</span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>range of scenarios, they tend to be less powerful, meaning they are less</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>likely to detect a true difference when one exists.</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a><span class="fu">### Independent vs. Paired Samples</span></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>A crucial consideration when conducting tests is whether the values in</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>one group are independent of those in the other. Non-independence occurs</span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>when the two measures are linked in some way—for instance, if they come</span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>from the same individual, time, or location.</span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>For example, when evaluating a treatment for high blood pressure, we</span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>might measure blood pressure before and after treatment on the same</span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>individuals. In this case, the before and after measurements are not</span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a>independent. If pairs of observations across groups share a common</span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>factor that makes them more similar to each other than to other</span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>observations, the samples are not independent.</span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a>We use different testing approaches for independent and non-independent</span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a>samples to account for this dependency.</span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a><span class="fu">### *T*-tests</span></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>A linear model with one explanatory variable with two groups is also</span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a>known as a **two-sample *t*-test** when the samples are independent and</span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a>as a **paired-samples *t*-test** when they are not. R does have a</span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a><span class="in">`t.test()`</span> function which allows you to fit a linear model with just two</span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a>groups. However, here we teach you to use and interpret the <span class="in">`lm()`</span></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a>function because it is more generalisable. You can use <span class="in">`lm()`</span> when you</span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a>have three or more groups or additional explanatory variables. The</span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a>output of <span class="in">`lm()`</span> is also in the same form as many other statistical</span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a>functions in R. This means what you learn in performing *t*-tests with</span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a><span class="in">`lm()`</span> will help you learn other methods more easily. However, it is</span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a>definitely not wrong to use <span class="in">`t.test()`</span> rather than <span class="in">`lm()`</span> for two-group</span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a>situations - the procedures are identical and the *p*-values will be the</span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a>same.</span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a><span class="fu">### Model assumptions</span></span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a>The assumptions of the general linear model are that the residuals are</span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a>normally distributed and have homogeneity of variance. A residual is the</span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a>difference between the predicted value and the observed value.</span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a>If we have a continuous response and a categorical explanatory variable</span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a>with two groups, we usually apply the general linear model with <span class="in">`lm()`</span></span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a>and *then* check the assumptions, however, we can sometimes tell when a</span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a>non-parametric test would be more appropriate before that:</span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Use common sense - the response should be continuous (or nearly</span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a>    continuous, see [Ideas about data: Theory and</span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a>    practice](ideas_about_data.html#theory-and-practice)). Consider</span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a>    whether you would expect the response to be continuous.</span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>We expect decimal places and few repeated values.</span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a>To examine the assumptions after fitting the linear model, we plot the</span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a>residuals and test them against the normal distribution in the same way</span>
<span id="cb28-90"><a href="#cb28-90" aria-hidden="true" tabindex="-1"></a>as we did for single linear regression.</span>
<span id="cb28-91"><a href="#cb28-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-92"><a href="#cb28-92" aria-hidden="true" tabindex="-1"></a><span class="fu">### Reporting</span></span>
<span id="cb28-93"><a href="#cb28-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-94"><a href="#cb28-94" aria-hidden="true" tabindex="-1"></a>In reporting the result of two-sample test we give:</span>
<span id="cb28-95"><a href="#cb28-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-96"><a href="#cb28-96" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>the significance of effect - whether there is there a difference</span>
<span id="cb28-97"><a href="#cb28-97" aria-hidden="true" tabindex="-1"></a>    between the groups</span>
<span id="cb28-98"><a href="#cb28-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-99"><a href="#cb28-99" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>parametric: whether there is there a difference between the</span>
<span id="cb28-100"><a href="#cb28-100" aria-hidden="true" tabindex="-1"></a>        groups means</span>
<span id="cb28-101"><a href="#cb28-101" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>non-parametric: whether there is there a difference between the</span>
<span id="cb28-102"><a href="#cb28-102" aria-hidden="true" tabindex="-1"></a>        group medians</span>
<span id="cb28-103"><a href="#cb28-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-104"><a href="#cb28-104" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>the direction of effect - which of the means/medians is greater</span>
<span id="cb28-105"><a href="#cb28-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-106"><a href="#cb28-106" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>the magnitude of effect - how big is the difference between the</span>
<span id="cb28-107"><a href="#cb28-107" aria-hidden="true" tabindex="-1"></a>    means/medians</span>
<span id="cb28-108"><a href="#cb28-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-109"><a href="#cb28-109" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>parametric: the means and standard errors for each group or the</span>
<span id="cb28-110"><a href="#cb28-110" aria-hidden="true" tabindex="-1"></a>        mean difference for paired samples</span>
<span id="cb28-111"><a href="#cb28-111" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>non-parametric: the medians for each group or the median</span>
<span id="cb28-112"><a href="#cb28-112" aria-hidden="true" tabindex="-1"></a>        difference for paired samples</span>
<span id="cb28-113"><a href="#cb28-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-114"><a href="#cb28-114" aria-hidden="true" tabindex="-1"></a>Figures should reflect what you have said in the statements. Ideally</span>
<span id="cb28-115"><a href="#cb28-115" aria-hidden="true" tabindex="-1"></a>they should show both the raw data and the statistical model:</span>
<span id="cb28-116"><a href="#cb28-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-117"><a href="#cb28-117" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>parametric: means and standard errors</span>
<span id="cb28-118"><a href="#cb28-118" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>non-parametric: boxplots with medians and interquartile range</span>
<span id="cb28-119"><a href="#cb28-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-120"><a href="#cb28-120" aria-hidden="true" tabindex="-1"></a>We will explore all of these ideas with some examples.</span>
<span id="cb28-121"><a href="#cb28-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-122"><a href="#cb28-122" aria-hidden="true" tabindex="-1"></a><span class="fu">## 🎬 Your turn!</span></span>
<span id="cb28-123"><a href="#cb28-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-124"><a href="#cb28-124" aria-hidden="true" tabindex="-1"></a>If you want to code along you will need to start a new [RStudio</span>
<span id="cb28-125"><a href="#cb28-125" aria-hidden="true" tabindex="-1"></a>project](workflow_rstudio.html#rstudio-projects), add a <span class="in">`data-raw`</span></span>
<span id="cb28-126"><a href="#cb28-126" aria-hidden="true" tabindex="-1"></a>folder and open a new script. You will also need to load the</span>
<span id="cb28-127"><a href="#cb28-127" aria-hidden="true" tabindex="-1"></a>**`tidyverse`** package <span class="co">[</span><span class="ot">@tidyverse</span><span class="co">]</span>.</span>
<span id="cb28-128"><a href="#cb28-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-129"><a href="#cb28-129" aria-hidden="true" tabindex="-1"></a><span class="fu">## Two independent samples, parametric</span></span>
<span id="cb28-130"><a href="#cb28-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-131"><a href="#cb28-131" aria-hidden="true" tabindex="-1"></a>A number of [subspecies of the common chaffinch, *Fringilla</span>
<span id="cb28-132"><a href="#cb28-132" aria-hidden="true" tabindex="-1"></a>coelebs*](https://en.wikipedia.org/wiki/Common_chaffinch), have been</span>
<span id="cb28-133"><a href="#cb28-133" aria-hidden="true" tabindex="-1"></a>described based principally on the differences in the pattern and colour</span>
<span id="cb28-134"><a href="#cb28-134" aria-hidden="true" tabindex="-1"></a>of the adult male plumage <span class="co">[</span><span class="ot">@suárez2009</span><span class="co">]</span>. Two of groups of these</span>
<span id="cb28-135"><a href="#cb28-135" aria-hidden="true" tabindex="-1"></a>subspecies are:</span>
<span id="cb28-136"><a href="#cb28-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-137"><a href="#cb28-137" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>the "coelebs group" (@fig-coelebs) that occurs in Europe and Asia</span>
<span id="cb28-138"><a href="#cb28-138" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>the "canariensis group" (@fig-canariensis) that occurs on the Canary</span>
<span id="cb28-139"><a href="#cb28-139" aria-hidden="true" tabindex="-1"></a>    Islands.</span>
<span id="cb28-140"><a href="#cb28-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-141"><a href="#cb28-141" aria-hidden="true" tabindex="-1"></a>::: {#fig-subspecies layout-ncol="2"}</span>
<span id="cb28-142"><a href="#cb28-142" aria-hidden="true" tabindex="-1"></a>![*F. c.</span>
<span id="cb28-143"><a href="#cb28-143" aria-hidden="true" tabindex="-1"></a>coelebs*](images/512px-Chaffinch_(Fringilla_coelebs).jpg){#fig-coelebs}</span>
<span id="cb28-144"><a href="#cb28-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-145"><a href="#cb28-145" aria-hidden="true" tabindex="-1"></a>![*F. c.</span>
<span id="cb28-146"><a href="#cb28-146" aria-hidden="true" tabindex="-1"></a>palmae*](images/512px-Fringilla_coelebs_palmae_-_Los_Tilos.jpg){#fig-canariensis}</span>
<span id="cb28-147"><a href="#cb28-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-148"><a href="#cb28-148" aria-hidden="true" tabindex="-1"></a>Adult male *Fringilla coelebs* of the coelebs group on the left (Andreas</span>
<span id="cb28-149"><a href="#cb28-149" aria-hidden="true" tabindex="-1"></a>Trepte, CC BY-SA 2.5 <span class="ot">&lt;https://creativecommons.org/licenses/by-sa/2.5&gt;</span>,</span>
<span id="cb28-150"><a href="#cb28-150" aria-hidden="true" tabindex="-1"></a>via Wikimedia Commons) and of the canariensis group on the right (H.</span>
<span id="cb28-151"><a href="#cb28-151" aria-hidden="true" tabindex="-1"></a>Zell, CC BY-SA 3.0 <span class="ot">&lt;https://creativecommons.org/licenses/by-sa/3.0&gt;</span>, via</span>
<span id="cb28-152"><a href="#cb28-152" aria-hidden="true" tabindex="-1"></a>Wikimedia Commons).</span>
<span id="cb28-153"><a href="#cb28-153" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb28-154"><a href="#cb28-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-155"><a href="#cb28-155" aria-hidden="true" tabindex="-1"></a>The data in <span class="co">[</span><span class="ot">chaff.txt</span><span class="co">](data-raw/chaff.txt)</span> give the masses of twenty</span>
<span id="cb28-156"><a href="#cb28-156" aria-hidden="true" tabindex="-1"></a>individuals from each subspecies. We want to know if the subspecies</span>
<span id="cb28-157"><a href="#cb28-157" aria-hidden="true" tabindex="-1"></a>differ in mass. These groups are independent - there is no link between</span>
<span id="cb28-158"><a href="#cb28-158" aria-hidden="true" tabindex="-1"></a>values in one group and any value in the other group. In this scenario</span>
<span id="cb28-159"><a href="#cb28-159" aria-hidden="true" tabindex="-1"></a>our null hypothesis, $H_0$, is that there is no difference between the</span>
<span id="cb28-160"><a href="#cb28-160" aria-hidden="true" tabindex="-1"></a>two subspecies in mass or that subspecies has no effect on mass. This is</span>
<span id="cb28-161"><a href="#cb28-161" aria-hidden="true" tabindex="-1"></a>written as: $H_0: \beta_1 = 0$.</span>
<span id="cb28-162"><a href="#cb28-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-163"><a href="#cb28-163" aria-hidden="true" tabindex="-1"></a><span class="fu">### Import and explore</span></span>
<span id="cb28-164"><a href="#cb28-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-165"><a href="#cb28-165" aria-hidden="true" tabindex="-1"></a>Import the data:</span>
<span id="cb28-166"><a href="#cb28-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-169"><a href="#cb28-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-170"><a href="#cb28-170" aria-hidden="true" tabindex="-1"></a>chaff <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">"data-raw/chaff.txt"</span>)</span>
<span id="cb28-171"><a href="#cb28-171" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-172"><a href="#cb28-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-175"><a href="#cb28-175" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-176"><a href="#cb28-176" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb28-177"><a href="#cb28-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-178"><a href="#cb28-178" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(chaff) <span class="sc">|&gt;</span> </span>
<span id="cb28-179"><a href="#cb28-179" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>() <span class="sc">|&gt;</span> </span>
<span id="cb28-180"><a href="#cb28-180" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">scroll_box</span>(<span class="at">height =</span> <span class="st">"200px"</span>)</span>
<span id="cb28-181"><a href="#cb28-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-182"><a href="#cb28-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-183"><a href="#cb28-183" aria-hidden="true" tabindex="-1"></a>These data are in tidy format <span class="co">[</span><span class="ot">@Wickham2014-nl</span><span class="co">]</span> - all the mass values</span>
<span id="cb28-184"><a href="#cb28-184" aria-hidden="true" tabindex="-1"></a>are in one column with another column indicating the subspecies. This</span>
<span id="cb28-185"><a href="#cb28-185" aria-hidden="true" tabindex="-1"></a>means they are well formatted for analysis and plotting.</span>
<span id="cb28-186"><a href="#cb28-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-187"><a href="#cb28-187" aria-hidden="true" tabindex="-1"></a>In the first instance, it is always sensible to create a rough plot of</span>
<span id="cb28-188"><a href="#cb28-188" aria-hidden="true" tabindex="-1"></a>our data. This is to give us an overview and help identify if there are</span>
<span id="cb28-189"><a href="#cb28-189" aria-hidden="true" tabindex="-1"></a>any issues like missing or extreme values. It also gives us idea what we</span>
<span id="cb28-190"><a href="#cb28-190" aria-hidden="true" tabindex="-1"></a>are expecting from the analysis which will make it easier for us to</span>
<span id="cb28-191"><a href="#cb28-191" aria-hidden="true" tabindex="-1"></a>identify if we make some mistake in applying that analysis.</span>
<span id="cb28-192"><a href="#cb28-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-193"><a href="#cb28-193" aria-hidden="true" tabindex="-1"></a>Violin plots (<span class="in">`geom_violin()`</span>), box plots (<span class="in">`geom_boxplot()`</span>, see</span>
<span id="cb28-194"><a href="#cb28-194" aria-hidden="true" tabindex="-1"></a>@fig-chaff-rough) or scatter plots (<span class="in">`geom_point()`</span>) all make good</span>
<span id="cb28-195"><a href="#cb28-195" aria-hidden="true" tabindex="-1"></a>choices for exploratory plotting and it does not matter which of these</span>
<span id="cb28-196"><a href="#cb28-196" aria-hidden="true" tabindex="-1"></a>you choose.</span>
<span id="cb28-197"><a href="#cb28-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-200"><a href="#cb28-200" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-201"><a href="#cb28-201" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-chaff-rough</span></span>
<span id="cb28-202"><a href="#cb28-202" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "The mass of two subspecies of chaffinch. A boxplot is a useful way to get an overview of the data and helps us identify any issues such as missing or extreme values. It also tells us what to expect from the analysis."</span></span>
<span id="cb28-203"><a href="#cb28-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-204"><a href="#cb28-204" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> chaff,</span>
<span id="cb28-205"><a href="#cb28-205" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> subspecies, <span class="at">y =</span> mass)) <span class="sc">+</span></span>
<span id="cb28-206"><a href="#cb28-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb28-207"><a href="#cb28-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-208"><a href="#cb28-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-209"><a href="#cb28-209" aria-hidden="true" tabindex="-1"></a>R will order the groups alphabetically by default.</span>
<span id="cb28-210"><a href="#cb28-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-211"><a href="#cb28-211" aria-hidden="true" tabindex="-1"></a>The figure suggests that the canariensis group is heavier than the</span>
<span id="cb28-212"><a href="#cb28-212" aria-hidden="true" tabindex="-1"></a>coelebs group.</span>
<span id="cb28-213"><a href="#cb28-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-214"><a href="#cb28-214" aria-hidden="true" tabindex="-1"></a>Summarising the data for each subspecies group is the next sensible</span>
<span id="cb28-215"><a href="#cb28-215" aria-hidden="true" tabindex="-1"></a>step. The most useful summary statistics are the means, standard</span>
<span id="cb28-216"><a href="#cb28-216" aria-hidden="true" tabindex="-1"></a>deviations, sample sizes and standard errors. I recommend the</span>
<span id="cb28-217"><a href="#cb28-217" aria-hidden="true" tabindex="-1"></a><span class="in">`group_by()`</span> and <span class="in">`summarise()`</span> approach:</span>
<span id="cb28-218"><a href="#cb28-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-221"><a href="#cb28-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-222"><a href="#cb28-222" aria-hidden="true" tabindex="-1"></a>chaff_summary <span class="ot">&lt;-</span> chaff <span class="sc">|&gt;</span> </span>
<span id="cb28-223"><a href="#cb28-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(subspecies) <span class="sc">|&gt;</span> </span>
<span id="cb28-224"><a href="#cb28-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(mass),</span>
<span id="cb28-225"><a href="#cb28-225" aria-hidden="true" tabindex="-1"></a>            <span class="at">std =</span> <span class="fu">sd</span>(mass),</span>
<span id="cb28-226"><a href="#cb28-226" aria-hidden="true" tabindex="-1"></a>            <span class="at">n =</span> <span class="fu">length</span>(mass),</span>
<span id="cb28-227"><a href="#cb28-227" aria-hidden="true" tabindex="-1"></a>            <span class="at">se =</span> std<span class="sc">/</span><span class="fu">sqrt</span>(n))</span>
<span id="cb28-228"><a href="#cb28-228" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-229"><a href="#cb28-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-230"><a href="#cb28-230" aria-hidden="true" tabindex="-1"></a>We have save the results to <span class="in">`chaff_summary`</span> so that we can use the means</span>
<span id="cb28-231"><a href="#cb28-231" aria-hidden="true" tabindex="-1"></a>and standard errors in our plot later.</span>
<span id="cb28-232"><a href="#cb28-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-235"><a href="#cb28-235" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-236"><a href="#cb28-236" aria-hidden="true" tabindex="-1"></a>chaff_summary</span>
<span id="cb28-237"><a href="#cb28-237" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-238"><a href="#cb28-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-239"><a href="#cb28-239" aria-hidden="true" tabindex="-1"></a><span class="fu">### Apply `lm()`</span></span>
<span id="cb28-240"><a href="#cb28-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-241"><a href="#cb28-241" aria-hidden="true" tabindex="-1"></a>We can create a two-sample model like this:</span>
<span id="cb28-242"><a href="#cb28-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-245"><a href="#cb28-245" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-246"><a href="#cb28-246" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">data =</span> chaff, mass <span class="sc">~</span> subspecies)</span>
<span id="cb28-247"><a href="#cb28-247" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-248"><a href="#cb28-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-249"><a href="#cb28-249" aria-hidden="true" tabindex="-1"></a>And examine the model with:</span>
<span id="cb28-250"><a href="#cb28-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-253"><a href="#cb28-253" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-254"><a href="#cb28-254" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod)</span>
<span id="cb28-255"><a href="#cb28-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-256"><a href="#cb28-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-257"><a href="#cb28-257" aria-hidden="true" tabindex="-1"></a>The Estimates in the Coefficients table give:</span>
<span id="cb28-258"><a href="#cb28-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-259"><a href="#cb28-259" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`(Intercept)`</span> known as $\beta_0$. The mean of the canariensis group</span>
<span id="cb28-260"><a href="#cb28-260" aria-hidden="true" tabindex="-1"></a>    (@fig-two-sample-lm-model). Just as the intercept is the value of</span>
<span id="cb28-261"><a href="#cb28-261" aria-hidden="true" tabindex="-1"></a>    the *y* (the response) when the value of *x* (the explanatory) is</span>
<span id="cb28-262"><a href="#cb28-262" aria-hidden="true" tabindex="-1"></a>    zero in a simple linear regression, this is the value of <span class="in">`mass`</span> when</span>
<span id="cb28-263"><a href="#cb28-263" aria-hidden="true" tabindex="-1"></a>    the <span class="in">`subspecies`</span> is at its first level. The order of the levels is</span>
<span id="cb28-264"><a href="#cb28-264" aria-hidden="true" tabindex="-1"></a>    alphabetical by default.</span>
<span id="cb28-265"><a href="#cb28-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-266"><a href="#cb28-266" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`subspeciescoelebs`</span>, known as $\beta_1$, is what needs to be added</span>
<span id="cb28-267"><a href="#cb28-267" aria-hidden="true" tabindex="-1"></a>    to the mean of the canariensis group to get the mean of the coelebs</span>
<span id="cb28-268"><a href="#cb28-268" aria-hidden="true" tabindex="-1"></a>    group (@fig-two-sample-lm-model). Just as the slope is amount of *y*</span>
<span id="cb28-269"><a href="#cb28-269" aria-hidden="true" tabindex="-1"></a>    that needs to be added for each unit of *x* in a simple linear</span>
<span id="cb28-270"><a href="#cb28-270" aria-hidden="true" tabindex="-1"></a>    regression, this is the amount of <span class="in">`mass`</span> that needs to be added when</span>
<span id="cb28-271"><a href="#cb28-271" aria-hidden="true" tabindex="-1"></a>    the <span class="in">`subspecies`</span> goes from its first level to its second level</span>
<span id="cb28-272"><a href="#cb28-272" aria-hidden="true" tabindex="-1"></a>    (*i.e.*, one unit). The <span class="in">`subspeciescoelebs`</span> estimate is negative so</span>
<span id="cb28-273"><a href="#cb28-273" aria-hidden="true" tabindex="-1"></a>    the the coelebs group mean is lower than the canariensis group mean</span>
<span id="cb28-274"><a href="#cb28-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-275"><a href="#cb28-275" aria-hidden="true" tabindex="-1"></a>The *p*-values on each line are tests of whether that coefficient is</span>
<span id="cb28-276"><a href="#cb28-276" aria-hidden="true" tabindex="-1"></a>different from zero. Thus it is:</span>
<span id="cb28-277"><a href="#cb28-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-278"><a href="#cb28-278" aria-hidden="true" tabindex="-1"></a><span class="in">`subspeciescoelebs  -1.7950     0.6781  -2.647   0.0118 *`</span></span>
<span id="cb28-279"><a href="#cb28-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-280"><a href="#cb28-280" aria-hidden="true" tabindex="-1"></a>that tells us the difference between the means is significant.</span>
<span id="cb28-281"><a href="#cb28-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-282"><a href="#cb28-282" aria-hidden="true" tabindex="-1"></a>The *F* value and *p*-value in the last line are a test of whether the</span>
<span id="cb28-283"><a href="#cb28-283" aria-hidden="true" tabindex="-1"></a>model as a whole explains a significant amount of variation in the</span>
<span id="cb28-284"><a href="#cb28-284" aria-hidden="true" tabindex="-1"></a>response variable. For a two-sample test, just like a regression, this</span>
<span id="cb28-285"><a href="#cb28-285" aria-hidden="true" tabindex="-1"></a>is exactly equivalent to the test of the slope against zero and the two</span>
<span id="cb28-286"><a href="#cb28-286" aria-hidden="true" tabindex="-1"></a>*p*-values will be the same.</span>
<span id="cb28-287"><a href="#cb28-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-290"><a href="#cb28-290" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-291"><a href="#cb28-291" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb28-292"><a href="#cb28-292" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-two-sample-lm-model</span></span>
<span id="cb28-293"><a href="#cb28-293" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "In a two-sample linear model, the first estimate is the intercept which is the mean of the first group. The second estimate is the 'slope' which is what has to added to the intercept to get the second group mean. Note that y axis starts at 15 to create more space for the annotations."</span></span>
<span id="cb28-294"><a href="#cb28-294" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb28-295"><a href="#cb28-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> chaff_summary, </span>
<span id="cb28-296"><a href="#cb28-296" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> subspecies, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean), </span>
<span id="cb28-297"><a href="#cb28-297" aria-hidden="true" tabindex="-1"></a>                <span class="at">colour =</span> pal3[<span class="dv">1</span>], <span class="at">linewidth =</span> <span class="dv">1</span>,</span>
<span id="cb28-298"><a href="#cb28-298" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb28-299"><a href="#cb28-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb28-300"><a href="#cb28-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb28-301"><a href="#cb28-301" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">30</span>),</span>
<span id="cb28-302"><a href="#cb28-302" aria-hidden="true" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">"Mass (g)"</span>) <span class="sc">+</span></span>
<span id="cb28-303"><a href="#cb28-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">0.8</span>,</span>
<span id="cb28-304"><a href="#cb28-304" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xend =</span> <span class="sc">-</span><span class="cn">Inf</span>,</span>
<span id="cb28-305"><a href="#cb28-305" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">3</span>,</span>
<span id="cb28-306"><a href="#cb28-306" aria-hidden="true" tabindex="-1"></a>                   <span class="at">yend =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>]),</span>
<span id="cb28-307"><a href="#cb28-307" aria-hidden="true" tabindex="-1"></a>               <span class="at">colour =</span> pal3[<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb28-308"><a href="#cb28-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>,</span>
<span id="cb28-309"><a href="#cb28-309" aria-hidden="true" tabindex="-1"></a>           <span class="at">x =</span> <span class="fl">1.1</span>, </span>
<span id="cb28-310"><a href="#cb28-310" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">4</span>,</span>
<span id="cb28-311"><a href="#cb28-311" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"Intercept (β0) is mean of</span><span class="sc">\n</span><span class="st">{ chaff_summary$subspecies[1] }: { mod$coefficients[1] |&gt; round(2) }"</span>), </span>
<span id="cb28-312"><a href="#cb28-312" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> pal3[<span class="dv">2</span>],</span>
<span id="cb28-313"><a href="#cb28-313" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb28-314"><a href="#cb28-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.5</span>,</span>
<span id="cb28-315"><a href="#cb28-315" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xend =</span> <span class="fl">1.5</span>,</span>
<span id="cb28-316"><a href="#cb28-316" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>],</span>
<span id="cb28-317"><a href="#cb28-317" aria-hidden="true" tabindex="-1"></a>                   <span class="at">yend =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>] <span class="sc">+</span> mod<span class="sc">$</span>coefficients[<span class="dv">2</span>]),</span>
<span id="cb28-318"><a href="#cb28-318" aria-hidden="true" tabindex="-1"></a>               <span class="at">colour =</span> pal3[<span class="dv">3</span>],</span>
<span id="cb28-319"><a href="#cb28-319" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.03</span>, <span class="st">"npc"</span>), </span>
<span id="cb28-320"><a href="#cb28-320" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ends =</span> <span class="st">"both"</span>)) <span class="sc">+</span></span>
<span id="cb28-321"><a href="#cb28-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.7</span>,</span>
<span id="cb28-322"><a href="#cb28-322" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xend =</span> <span class="fl">1.5</span>,</span>
<span id="cb28-323"><a href="#cb28-323" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">3</span>,</span>
<span id="cb28-324"><a href="#cb28-324" aria-hidden="true" tabindex="-1"></a>                   <span class="at">yend =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>] <span class="sc">-</span> <span class="dv">1</span>),</span>
<span id="cb28-325"><a href="#cb28-325" aria-hidden="true" tabindex="-1"></a>               <span class="at">colour =</span> pal3[<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb28-326"><a href="#cb28-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>,</span>
<span id="cb28-327"><a href="#cb28-327" aria-hidden="true" tabindex="-1"></a>           <span class="at">x =</span> <span class="dv">2</span>, </span>
<span id="cb28-328"><a href="#cb28-328" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> mod<span class="sc">$</span>coefficients[<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">5</span>,</span>
<span id="cb28-329"><a href="#cb28-329" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"subspeciescoelebs (β1) is the</span><span class="sc">\n</span><span class="st">difference between mean of</span><span class="sc">\n</span><span class="st">{ chaff_summary$subspecies[1] } and { chaff_summary$subspecies[2] }: { mod$coefficients[2] |&gt; round(2) }"</span>), </span>
<span id="cb28-330"><a href="#cb28-330" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> pal3[<span class="dv">2</span>],</span>
<span id="cb28-331"><a href="#cb28-331" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb28-332"><a href="#cb28-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb28-333"><a href="#cb28-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-334"><a href="#cb28-334" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-335"><a href="#cb28-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-336"><a href="#cb28-336" aria-hidden="true" tabindex="-1"></a><span class="fu">### Check assumptions</span></span>
<span id="cb28-337"><a href="#cb28-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-338"><a href="#cb28-338" aria-hidden="true" tabindex="-1"></a>Check the assumptions: All general linear models assume the "residuals"</span>
<span id="cb28-339"><a href="#cb28-339" aria-hidden="true" tabindex="-1"></a>are normally distributed and have "homogeneity" of variance.</span>
<span id="cb28-340"><a href="#cb28-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-341"><a href="#cb28-341" aria-hidden="true" tabindex="-1"></a>Our first check of these assumptions is to use common sense: mass is a</span>
<span id="cb28-342"><a href="#cb28-342" aria-hidden="true" tabindex="-1"></a>continuous variable and we would expect it to be normally distributed</span>
<span id="cb28-343"><a href="#cb28-343" aria-hidden="true" tabindex="-1"></a>thus we would also expect the residuals to be normally distributed.</span>
<span id="cb28-344"><a href="#cb28-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-345"><a href="#cb28-345" aria-hidden="true" tabindex="-1"></a>We then plot the residuals. The <span class="in">`plot()`</span> function can be used to plot</span>
<span id="cb28-346"><a href="#cb28-346" aria-hidden="true" tabindex="-1"></a>the residuals against the fitted values (See @fig-lm-plot1). This is a</span>
<span id="cb28-347"><a href="#cb28-347" aria-hidden="true" tabindex="-1"></a>good way to check for homogeneity of variance.</span>
<span id="cb28-348"><a href="#cb28-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-351"><a href="#cb28-351" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-352"><a href="#cb28-352" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-lm-plot1</span></span>
<span id="cb28-353"><a href="#cb28-353" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "A plot of the residuals against the fitted values shows the points are distributed similarly in each group. This is a good sign for the assumption of homogeneity of variance."</span></span>
<span id="cb28-354"><a href="#cb28-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-355"><a href="#cb28-355" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mod, <span class="at">which =</span> <span class="dv">1</span>)</span>
<span id="cb28-356"><a href="#cb28-356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-357"><a href="#cb28-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-358"><a href="#cb28-358" aria-hidden="true" tabindex="-1"></a>We can also use a histogram to check for normality (See @fig-lm-plot2).</span>
<span id="cb28-359"><a href="#cb28-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-362"><a href="#cb28-362" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-363"><a href="#cb28-363" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-lm-plot2</span></span>
<span id="cb28-364"><a href="#cb28-364" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "A histogram of residuals is symetrical and seems consistent with a normal distribution. This is a good sign for the assumption of normally distributed residuals."</span></span>
<span id="cb28-365"><a href="#cb28-365" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> mod<span class="sc">$</span>residuals)) <span class="sc">+</span> </span>
<span id="cb28-366"><a href="#cb28-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>)</span>
<span id="cb28-367"><a href="#cb28-367" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-368"><a href="#cb28-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-369"><a href="#cb28-369" aria-hidden="true" tabindex="-1"></a>Finally, we can use the Shapiro-Wilk test to test for normality.</span>
<span id="cb28-370"><a href="#cb28-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-373"><a href="#cb28-373" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-374"><a href="#cb28-374" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(mod<span class="sc">$</span>residuals)</span>
<span id="cb28-375"><a href="#cb28-375" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-376"><a href="#cb28-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-377"><a href="#cb28-377" aria-hidden="true" tabindex="-1"></a>The p-value is greater than 0.05 so this test of the normality</span>
<span id="cb28-378"><a href="#cb28-378" aria-hidden="true" tabindex="-1"></a>assumption is not significant.</span>
<span id="cb28-379"><a href="#cb28-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-380"><a href="#cb28-380" aria-hidden="true" tabindex="-1"></a>Taken together, these results suggest that the assumptions of normality</span>
<span id="cb28-381"><a href="#cb28-381" aria-hidden="true" tabindex="-1"></a>and homogeneity of variance are not violated.</span>
<span id="cb28-382"><a href="#cb28-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-383"><a href="#cb28-383" aria-hidden="true" tabindex="-1"></a><span class="fu">### Report</span></span>
<span id="cb28-384"><a href="#cb28-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-385"><a href="#cb28-385" aria-hidden="true" tabindex="-1"></a>Canariensis chaffinches ($\bar{x} \pm s.e$: 22.48 $\pm$ 0.48) were</span>
<span id="cb28-386"><a href="#cb28-386" aria-hidden="true" tabindex="-1"></a>significantly heavier than Coelebs (20.28 $\pm$ 0.48 ) (*t* = 2.65;</span>
<span id="cb28-387"><a href="#cb28-387" aria-hidden="true" tabindex="-1"></a>*d.f.* = 38; *p* = 0.012). See @fig-chaff.</span>
<span id="cb28-388"><a href="#cb28-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-389"><a href="#cb28-389" aria-hidden="true" tabindex="-1"></a>::: {#fig-chaff}</span>
<span id="cb28-392"><a href="#cb28-392" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-393"><a href="#cb28-393" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb28-394"><a href="#cb28-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-395"><a href="#cb28-395" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb28-396"><a href="#cb28-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> chaff, <span class="fu">aes</span>(<span class="at">x =</span> subspecies, <span class="at">y =</span> mass),</span>
<span id="cb28-397"><a href="#cb28-397" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="dv">0</span>),</span>
<span id="cb28-398"><a href="#cb28-398" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="st">"gray50"</span>) <span class="sc">+</span></span>
<span id="cb28-399"><a href="#cb28-399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> chaff_summary, </span>
<span id="cb28-400"><a href="#cb28-400" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> subspecies, <span class="at">ymin =</span> mean <span class="sc">-</span> se, <span class="at">ymax =</span> mean <span class="sc">+</span> se),</span>
<span id="cb28-401"><a href="#cb28-401" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb28-402"><a href="#cb28-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> chaff_summary, </span>
<span id="cb28-403"><a href="#cb28-403" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> subspecies, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean),</span>
<span id="cb28-404"><a href="#cb28-404" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb28-405"><a href="#cb28-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Mass (g)"</span>, </span>
<span id="cb28-406"><a href="#cb28-406" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">30</span>), </span>
<span id="cb28-407"><a href="#cb28-407" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb28-408"><a href="#cb28-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Subspecies"</span>, </span>
<span id="cb28-409"><a href="#cb28-409" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Canariensis"</span>, <span class="st">"Coelebs"</span>)) <span class="sc">+</span></span>
<span id="cb28-410"><a href="#cb28-410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb28-411"><a href="#cb28-411" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="dv">28</span>, <span class="at">yend =</span> <span class="dv">28</span>,</span>
<span id="cb28-412"><a href="#cb28-412" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb28-413"><a href="#cb28-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="dv">29</span>, </span>
<span id="cb28-414"><a href="#cb28-414" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(p)<span class="sc">~</span><span class="st">"= 0.012"</span>)) <span class="sc">+</span></span>
<span id="cb28-415"><a href="#cb28-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb28-416"><a href="#cb28-416" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-417"><a href="#cb28-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-418"><a href="#cb28-418" aria-hidden="true" tabindex="-1"></a>**Canariensis chaffinches are heavier than Coelebs chaffinches**. The</span>
<span id="cb28-419"><a href="#cb28-419" aria-hidden="true" tabindex="-1"></a>mean mass of 20 randomly sampled males from each subspecies was</span>
<span id="cb28-420"><a href="#cb28-420" aria-hidden="true" tabindex="-1"></a>determined. Error bars are $\pm$ 1 standard error. Canariensis</span>
<span id="cb28-421"><a href="#cb28-421" aria-hidden="true" tabindex="-1"></a>chaffinches were significantly heavier than Coelebs (*t* = 2.65; *d.f.*</span>
<span id="cb28-422"><a href="#cb28-422" aria-hidden="true" tabindex="-1"></a>= 38; *p* = 0.012). Data analysis was conducted in R <span class="co">[</span><span class="ot">@R-core</span><span class="co">]</span> with</span>
<span id="cb28-423"><a href="#cb28-423" aria-hidden="true" tabindex="-1"></a>tidyverse packages <span class="co">[</span><span class="ot">@tidyverse</span><span class="co">]</span>.</span>
<span id="cb28-424"><a href="#cb28-424" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb28-425"><a href="#cb28-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-426"><a href="#cb28-426" aria-hidden="true" tabindex="-1"></a><span class="fu">## Two independent samples, non-parametric</span></span>
<span id="cb28-427"><a href="#cb28-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-428"><a href="#cb28-428" aria-hidden="true" tabindex="-1"></a>The non-parametric equivalent of the linear model with two independent</span>
<span id="cb28-429"><a href="#cb28-429" aria-hidden="true" tabindex="-1"></a>samples is the "Wilcoxon rank sum test" <span class="co">[</span><span class="ot">@wilcoxon1945</span><span class="co">]</span>. It is commonly</span>
<span id="cb28-430"><a href="#cb28-430" aria-hidden="true" tabindex="-1"></a>also known as the Mann-Whitney or Wilcoxon–Mann–Whitney.</span>
<span id="cb28-431"><a href="#cb28-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-432"><a href="#cb28-432" aria-hidden="true" tabindex="-1"></a>The general question you have about your data - are these two groups</span>
<span id="cb28-433"><a href="#cb28-433" aria-hidden="true" tabindex="-1"></a>different - is the same, but one of more of the following is true:</span>
<span id="cb28-434"><a href="#cb28-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-435"><a href="#cb28-435" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>the response variable is not continuous</span>
<span id="cb28-436"><a href="#cb28-436" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>the residuals are not normally distributed</span>
<span id="cb28-437"><a href="#cb28-437" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>the sample size is too small to tell if they are normally</span>
<span id="cb28-438"><a href="#cb28-438" aria-hidden="true" tabindex="-1"></a>    distributed.</span>
<span id="cb28-439"><a href="#cb28-439" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>the variance is not homogeneous</span>
<span id="cb28-440"><a href="#cb28-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-441"><a href="#cb28-441" aria-hidden="true" tabindex="-1"></a>The test is a applied in R with the <span class="in">`wilcox.test()`</span> function.</span>
<span id="cb28-442"><a href="#cb28-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-443"><a href="#cb28-443" aria-hidden="true" tabindex="-1"></a>The data in <span class="co">[</span><span class="ot">arabidopsis.txt</span><span class="co">](data-raw/arabidopsis.txt)</span> give the number</span>
<span id="cb28-444"><a href="#cb28-444" aria-hidden="true" tabindex="-1"></a>of leaves on eight wildtype and eight mutant *Arabidopsis thaliana*</span>
<span id="cb28-445"><a href="#cb28-445" aria-hidden="true" tabindex="-1"></a>plants. We want to know if the two types of plants have differing</span>
<span id="cb28-446"><a href="#cb28-446" aria-hidden="true" tabindex="-1"></a>numbers of leaves. These are counts, so they are not continuous and the</span>
<span id="cb28-447"><a href="#cb28-447" aria-hidden="true" tabindex="-1"></a>sample sizes are quite small. A non-parametric test is a safer option.</span>
<span id="cb28-448"><a href="#cb28-448" aria-hidden="true" tabindex="-1"></a>In this scenario our null hypothesis, $H_0$, is that there is no</span>
<span id="cb28-449"><a href="#cb28-449" aria-hidden="true" tabindex="-1"></a>difference between the two types of plant in the number of leaves.</span>
<span id="cb28-450"><a href="#cb28-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-451"><a href="#cb28-451" aria-hidden="true" tabindex="-1"></a><span class="fu">### Import and explore</span></span>
<span id="cb28-452"><a href="#cb28-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-455"><a href="#cb28-455" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-456"><a href="#cb28-456" aria-hidden="true" tabindex="-1"></a>arabidopsis <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">"data-raw/arabidopsis.txt"</span>)</span>
<span id="cb28-457"><a href="#cb28-457" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-458"><a href="#cb28-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-459"><a href="#cb28-459" aria-hidden="true" tabindex="-1"></a>These data are in tidy format <span class="co">[</span><span class="ot">@Wickham2014-nl</span><span class="co">]</span> - the numbers of leaves</span>
<span id="cb28-460"><a href="#cb28-460" aria-hidden="true" tabindex="-1"></a>are in one column with another column indicating whether the observation</span>
<span id="cb28-461"><a href="#cb28-461" aria-hidden="true" tabindex="-1"></a>comes from a wildtype or mutant *Arabidopsis*. This means they are well</span>
<span id="cb28-462"><a href="#cb28-462" aria-hidden="true" tabindex="-1"></a>formatted for analysis and plotting.</span>
<span id="cb28-463"><a href="#cb28-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-464"><a href="#cb28-464" aria-hidden="true" tabindex="-1"></a>Create a quick plot of the data:</span>
<span id="cb28-465"><a href="#cb28-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-468"><a href="#cb28-468" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-469"><a href="#cb28-469" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-arabid-rough</span></span>
<span id="cb28-470"><a href="#cb28-470" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "The number of leaves on mutant and wildtype plants. A boxplot is a useful way to get an overview of the data and helps us identify any issues such as missing or extreme values. It also tells us what to expect from the analysis."</span></span>
<span id="cb28-471"><a href="#cb28-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-472"><a href="#cb28-472" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> arabidopsis, </span>
<span id="cb28-473"><a href="#cb28-473" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> type, <span class="at">y =</span> leaves)) <span class="sc">+</span></span>
<span id="cb28-474"><a href="#cb28-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb28-475"><a href="#cb28-475" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-476"><a href="#cb28-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-477"><a href="#cb28-477" aria-hidden="true" tabindex="-1"></a>Our rough plot shows that the mutant plants have fewer leaves than the</span>
<span id="cb28-478"><a href="#cb28-478" aria-hidden="true" tabindex="-1"></a>wildtype plants.</span>
<span id="cb28-479"><a href="#cb28-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-480"><a href="#cb28-480" aria-hidden="true" tabindex="-1"></a>Summarising the data using the median and interquartile range is more</span>
<span id="cb28-481"><a href="#cb28-481" aria-hidden="true" tabindex="-1"></a>aligned to the type of data and the type of analysis than using means</span>
<span id="cb28-482"><a href="#cb28-482" aria-hidden="true" tabindex="-1"></a>and standard deviations:</span>
<span id="cb28-483"><a href="#cb28-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-486"><a href="#cb28-486" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-487"><a href="#cb28-487" aria-hidden="true" tabindex="-1"></a>arabidopsis_summary <span class="ot">&lt;-</span> arabidopsis <span class="sc">|&gt;</span> </span>
<span id="cb28-488"><a href="#cb28-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(type) <span class="sc">|&gt;</span> </span>
<span id="cb28-489"><a href="#cb28-489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median =</span> <span class="fu">median</span>(leaves),</span>
<span id="cb28-490"><a href="#cb28-490" aria-hidden="true" tabindex="-1"></a>            <span class="at">interquartile  =</span> <span class="fu">IQR</span>(leaves),</span>
<span id="cb28-491"><a href="#cb28-491" aria-hidden="true" tabindex="-1"></a>            <span class="at">n =</span> <span class="fu">length</span>(leaves))</span>
<span id="cb28-492"><a href="#cb28-492" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-493"><a href="#cb28-493" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-494"><a href="#cb28-494" aria-hidden="true" tabindex="-1"></a>View the results:</span>
<span id="cb28-495"><a href="#cb28-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-498"><a href="#cb28-498" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-499"><a href="#cb28-499" aria-hidden="true" tabindex="-1"></a>arabidopsis_summary</span>
<span id="cb28-500"><a href="#cb28-500" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-501"><a href="#cb28-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-502"><a href="#cb28-502" aria-hidden="true" tabindex="-1"></a><span class="fu">### Apply `wilcox.test()`</span></span>
<span id="cb28-503"><a href="#cb28-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-504"><a href="#cb28-504" aria-hidden="true" tabindex="-1"></a>We pass the dataframe and variables to <span class="in">`wilcox.test()`</span> in the same way</span>
<span id="cb28-505"><a href="#cb28-505" aria-hidden="true" tabindex="-1"></a>as we did for <span class="in">`lm()`</span>. We give the data argument and a "formula" which</span>
<span id="cb28-506"><a href="#cb28-506" aria-hidden="true" tabindex="-1"></a>says <span class="in">`leaves ~ type`</span> meaning "explain leaves by type".</span>
<span id="cb28-507"><a href="#cb28-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-510"><a href="#cb28-510" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-511"><a href="#cb28-511" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(<span class="at">data =</span> arabidopsis, leaves <span class="sc">~</span> type)</span>
<span id="cb28-512"><a href="#cb28-512" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-513"><a href="#cb28-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-514"><a href="#cb28-514" aria-hidden="true" tabindex="-1"></a>The warning message "Warning: cannot compute exact p-value with ties" is</span>
<span id="cb28-515"><a href="#cb28-515" aria-hidden="true" tabindex="-1"></a>not something to worry about too much. It is a warning rather than an</span>
<span id="cb28-516"><a href="#cb28-516" aria-hidden="true" tabindex="-1"></a>indication that your results are incorrect. It means the *p* -value is</span>
<span id="cb28-517"><a href="#cb28-517" aria-hidden="true" tabindex="-1"></a>based on an approximation rather than being exact because there are ties</span>
<span id="cb28-518"><a href="#cb28-518" aria-hidden="true" tabindex="-1"></a>(some values are the same).</span>
<span id="cb28-519"><a href="#cb28-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-520"><a href="#cb28-520" aria-hidden="true" tabindex="-1"></a>The result of the test is given on this line:</span>
<span id="cb28-521"><a href="#cb28-521" aria-hidden="true" tabindex="-1"></a><span class="in">`W = 5, p-value = 0.005051`</span>. <span class="in">`W`</span> is the test statistic. The *p*-value is</span>
<span id="cb28-522"><a href="#cb28-522" aria-hidden="true" tabindex="-1"></a>less than 0.05 meaning there is a significant difference in the number</span>
<span id="cb28-523"><a href="#cb28-523" aria-hidden="true" tabindex="-1"></a>of leaves on wildtype and mutant plants.</span>
<span id="cb28-524"><a href="#cb28-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-525"><a href="#cb28-525" aria-hidden="true" tabindex="-1"></a><span class="fu">### Report</span></span>
<span id="cb28-526"><a href="#cb28-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-527"><a href="#cb28-527" aria-hidden="true" tabindex="-1"></a>There are significantly more leaves on wildtype (median = 8.5) than</span>
<span id="cb28-528"><a href="#cb28-528" aria-hidden="true" tabindex="-1"></a>mutant (median = 5) plants (Wilcoxon rank sum test: *W* = 5, $n_1$ = 8,</span>
<span id="cb28-529"><a href="#cb28-529" aria-hidden="true" tabindex="-1"></a>$n_2$ = 8, *p* = 0.005). See @fig-arabid.</span>
<span id="cb28-530"><a href="#cb28-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-531"><a href="#cb28-531" aria-hidden="true" tabindex="-1"></a>::: {#fig-arabid}</span>
<span id="cb28-534"><a href="#cb28-534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-535"><a href="#cb28-535" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb28-536"><a href="#cb28-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-537"><a href="#cb28-537" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> arabidopsis, </span>
<span id="cb28-538"><a href="#cb28-538" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> type, <span class="at">y =</span> leaves)) <span class="sc">+</span></span>
<span id="cb28-539"><a href="#cb28-539" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb28-540"><a href="#cb28-540" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Number of leaves"</span>, </span>
<span id="cb28-541"><a href="#cb28-541" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>), </span>
<span id="cb28-542"><a href="#cb28-542" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb28-543"><a href="#cb28-543" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">""</span>, </span>
<span id="cb28-544"><a href="#cb28-544" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Mutatnt"</span>, <span class="st">"Wildtype"</span>)) <span class="sc">+</span></span>
<span id="cb28-545"><a href="#cb28-545" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb28-546"><a href="#cb28-546" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">10.5</span>, <span class="at">yend =</span> <span class="fl">10.5</span>,</span>
<span id="cb28-547"><a href="#cb28-547" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb28-548"><a href="#cb28-548" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="dv">11</span>, </span>
<span id="cb28-549"><a href="#cb28-549" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(p)<span class="sc">~</span><span class="st">"= 0.005"</span>)) <span class="sc">+</span></span>
<span id="cb28-550"><a href="#cb28-550" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb28-551"><a href="#cb28-551" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-552"><a href="#cb28-552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-553"><a href="#cb28-553" aria-hidden="true" tabindex="-1"></a>**Mutant *Arabidopsis thaliana* have fewer leaves**. There are</span>
<span id="cb28-554"><a href="#cb28-554" aria-hidden="true" tabindex="-1"></a>significantly more leaves on wildtype than mutant plants (Wilcoxon rank</span>
<span id="cb28-555"><a href="#cb28-555" aria-hidden="true" tabindex="-1"></a>sum test: *W* = 5, $n_1$ = 8, $n_2$ = 8, *p* = 0.005). The heavy lines</span>
<span id="cb28-556"><a href="#cb28-556" aria-hidden="true" tabindex="-1"></a>indicate the median of leaves, boxes indicate the interquartile range</span>
<span id="cb28-557"><a href="#cb28-557" aria-hidden="true" tabindex="-1"></a>and whiskers the range. Data analysis was conducted in R <span class="co">[</span><span class="ot">@R-core</span><span class="co">]</span> with</span>
<span id="cb28-558"><a href="#cb28-558" aria-hidden="true" tabindex="-1"></a>tidyverse packages <span class="co">[</span><span class="ot">@tidyverse</span><span class="co">]</span>.</span>
<span id="cb28-559"><a href="#cb28-559" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb28-560"><a href="#cb28-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-561"><a href="#cb28-561" aria-hidden="true" tabindex="-1"></a><span class="fu">## Two paired-samples, parametric</span></span>
<span id="cb28-562"><a href="#cb28-562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-563"><a href="#cb28-563" aria-hidden="true" tabindex="-1"></a>The data in <span class="co">[</span><span class="ot">marks.csv</span><span class="co">](data-raw/marks.csv)</span> give the marks for ten</span>
<span id="cb28-564"><a href="#cb28-564" aria-hidden="true" tabindex="-1"></a>students in two subjects: Data Analysis and Biology. These data are</span>
<span id="cb28-565"><a href="#cb28-565" aria-hidden="true" tabindex="-1"></a>paired because we have two marks from one student so that a mark in one</span>
<span id="cb28-566"><a href="#cb28-566" aria-hidden="true" tabindex="-1"></a>group has a closer relationship with one of the marks in the other group</span>
<span id="cb28-567"><a href="#cb28-567" aria-hidden="true" tabindex="-1"></a>than with any of the other values. We want to know if students do</span>
<span id="cb28-568"><a href="#cb28-568" aria-hidden="true" tabindex="-1"></a>equally well in both subjects. In this scenario our null hypothesis,</span>
<span id="cb28-569"><a href="#cb28-569" aria-hidden="true" tabindex="-1"></a>$H_0$, is that there is no difference between the Data Analysis and</span>
<span id="cb28-570"><a href="#cb28-570" aria-hidden="true" tabindex="-1"></a>Biology marks for a student.</span>
<span id="cb28-571"><a href="#cb28-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-572"><a href="#cb28-572" aria-hidden="true" tabindex="-1"></a><span class="fu">### Import and explore</span></span>
<span id="cb28-573"><a href="#cb28-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-574"><a href="#cb28-574" aria-hidden="true" tabindex="-1"></a>Import the data:</span>
<span id="cb28-575"><a href="#cb28-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-578"><a href="#cb28-578" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-579"><a href="#cb28-579" aria-hidden="true" tabindex="-1"></a>marks <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data-raw/marks.csv"</span>)</span>
<span id="cb28-580"><a href="#cb28-580" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-581"><a href="#cb28-581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-582"><a href="#cb28-582" aria-hidden="true" tabindex="-1"></a>Since these data are paired, it makes sense to highlight how the marks</span>
<span id="cb28-583"><a href="#cb28-583" aria-hidden="true" tabindex="-1"></a>differ for each student. One way of doing that is to draw a line linking</span>
<span id="cb28-584"><a href="#cb28-584" aria-hidden="true" tabindex="-1"></a>their marks in each subject. This is known as a spaghetti plot. We can</span>
<span id="cb28-585"><a href="#cb28-585" aria-hidden="true" tabindex="-1"></a>use two geoms: <span class="in">`geom_point()`</span> and <span class="in">`geom_line()`</span>. To join a student's</span>
<span id="cb28-586"><a href="#cb28-586" aria-hidden="true" tabindex="-1"></a>marks, we need to set the <span class="in">`group`</span> aesthetic to</span>
<span id="cb28-587"><a href="#cb28-587" aria-hidden="true" tabindex="-1"></a><span class="in">`student`</span>.<span class="ot">[^two_sample_tests-1]</span></span>
<span id="cb28-588"><a href="#cb28-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-589"><a href="#cb28-589" aria-hidden="true" tabindex="-1"></a><span class="ot">[^two_sample_tests-1]: </span>You might like to try removing</span>
<span id="cb28-590"><a href="#cb28-590" aria-hidden="true" tabindex="-1"></a>    <span class="in">`aes(group = student)`</span> to see what ggplot does when the lines are</span>
<span id="cb28-591"><a href="#cb28-591" aria-hidden="true" tabindex="-1"></a>    *not* grouped by student.</span>
<span id="cb28-592"><a href="#cb28-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-595"><a href="#cb28-595" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-596"><a href="#cb28-596" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> marks, <span class="fu">aes</span>(<span class="at">x =</span> subject, <span class="at">y =</span> mark)) <span class="sc">+</span></span>
<span id="cb28-597"><a href="#cb28-597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb28-598"><a href="#cb28-598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> student))</span>
<span id="cb28-599"><a href="#cb28-599" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-600"><a href="#cb28-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-601"><a href="#cb28-601" aria-hidden="true" tabindex="-1"></a>Summarise the data so that we can use the means in plots later:</span>
<span id="cb28-602"><a href="#cb28-602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-605"><a href="#cb28-605" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-606"><a href="#cb28-606" aria-hidden="true" tabindex="-1"></a>marks_summary <span class="ot">&lt;-</span> marks <span class="sc">|&gt;</span></span>
<span id="cb28-607"><a href="#cb28-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(subject) <span class="sc">|&gt;</span></span>
<span id="cb28-608"><a href="#cb28-608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(mark))</span>
<span id="cb28-609"><a href="#cb28-609" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-610"><a href="#cb28-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-611"><a href="#cb28-611" aria-hidden="true" tabindex="-1"></a>A paired test requires us to into account the variation between</span>
<span id="cb28-612"><a href="#cb28-612" aria-hidden="true" tabindex="-1"></a>students.</span>
<span id="cb28-613"><a href="#cb28-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-614"><a href="#cb28-614" aria-hidden="true" tabindex="-1"></a><span class="fu">### Apply `lm()`</span></span>
<span id="cb28-615"><a href="#cb28-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-616"><a href="#cb28-616" aria-hidden="true" tabindex="-1"></a>We can create a paired-sample model with the <span class="in">`lm()`</span></span>
<span id="cb28-617"><a href="#cb28-617" aria-hidden="true" tabindex="-1"></a>function<span class="ot">[^two_sample_tests-2]</span> like this:</span>
<span id="cb28-618"><a href="#cb28-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-619"><a href="#cb28-619" aria-hidden="true" tabindex="-1"></a><span class="ot">[^two_sample_tests-2]: </span>This is not the only way to apply a paired test.</span>
<span id="cb28-620"><a href="#cb28-620" aria-hidden="true" tabindex="-1"></a>    When there are only two groups and no other explanatory variables,</span>
<span id="cb28-621"><a href="#cb28-621" aria-hidden="true" tabindex="-1"></a>    we can use <span class="in">`t.test(data = marks, mark ~ subject, paired = TRUE)`</span>. A</span>
<span id="cb28-622"><a href="#cb28-622" aria-hidden="true" tabindex="-1"></a>    more general method that works when you have two or more</span>
<span id="cb28-623"><a href="#cb28-623" aria-hidden="true" tabindex="-1"></a>    non-independent values (e.g., more than two subjects) or additional</span>
<span id="cb28-624"><a href="#cb28-624" aria-hidden="true" tabindex="-1"></a>    explanatory variables is to create a "linear mixed model" with</span>
<span id="cb28-625"><a href="#cb28-625" aria-hidden="true" tabindex="-1"></a>    <span class="in">`lmer()`</span>.</span>
<span id="cb28-626"><a href="#cb28-626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-629"><a href="#cb28-629" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-630"><a href="#cb28-630" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(mark <span class="sc">~</span> subject <span class="sc">+</span> <span class="fu">factor</span>(student), <span class="at">data =</span> marks)</span>
<span id="cb28-631"><a href="#cb28-631" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-632"><a href="#cb28-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-633"><a href="#cb28-633" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`mark`</span> is the dependent variable (response).</span>
<span id="cb28-634"><a href="#cb28-634" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`subject`</span> is the independent variable (explanatory factor).</span>
<span id="cb28-635"><a href="#cb28-635" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`factor(student)`</span> accounts for the pairing by treating student as</span>
<span id="cb28-636"><a href="#cb28-636" aria-hidden="true" tabindex="-1"></a>    another explanatory variable. We have used factor because the values</span>
<span id="cb28-637"><a href="#cb28-637" aria-hidden="true" tabindex="-1"></a>    in students are the numbers 1 to 10 and we want <span class="in">`student`</span> to be</span>
<span id="cb28-638"><a href="#cb28-638" aria-hidden="true" tabindex="-1"></a>    treated as a category not a number</span>
<span id="cb28-639"><a href="#cb28-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-642"><a href="#cb28-642" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-643"><a href="#cb28-643" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod)</span>
<span id="cb28-644"><a href="#cb28-644" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-645"><a href="#cb28-645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-646"><a href="#cb28-646" aria-hidden="true" tabindex="-1"></a>The coefficient for <span class="in">`(Intercept)`</span> gives the mean Biology mark and that</span>
<span id="cb28-647"><a href="#cb28-647" aria-hidden="true" tabindex="-1"></a>for <span class="in">`subjectDataAnalysis`</span> is amount that the Data Analysis mark are</span>
<span id="cb28-648"><a href="#cb28-648" aria-hidden="true" tabindex="-1"></a>above Biology marks in general. The *p*-value tests whether this</span>
<span id="cb28-649"><a href="#cb28-649" aria-hidden="true" tabindex="-1"></a>difference is significantly different from zero. The rest of the output</span>
<span id="cb28-650"><a href="#cb28-650" aria-hidden="true" tabindex="-1"></a>considers how students differ. You can ignore this here.</span>
<span id="cb28-651"><a href="#cb28-651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-652"><a href="#cb28-652" aria-hidden="true" tabindex="-1"></a>If you find this a bit overwhelming to read you can use the <span class="in">`anova()`</span></span>
<span id="cb28-653"><a href="#cb28-653" aria-hidden="true" tabindex="-1"></a>function on the model object to get a simpler output:</span>
<span id="cb28-654"><a href="#cb28-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-657"><a href="#cb28-657" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-658"><a href="#cb28-658" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(mod)</span>
<span id="cb28-659"><a href="#cb28-659" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-660"><a href="#cb28-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-661"><a href="#cb28-661" aria-hidden="true" tabindex="-1"></a>You will notice that the *p*-value for subject is the same. The test</span>
<span id="cb28-662"><a href="#cb28-662" aria-hidden="true" tabindex="-1"></a>statistic, $F$ has a value of </span>
<span id="cb28-663"><a href="#cb28-663" aria-hidden="true" tabindex="-1"></a><span class="in">`r round(anova(mod)$F[1],2)`</span> </span>
<span id="cb28-664"><a href="#cb28-664" aria-hidden="true" tabindex="-1"></a>and degrees of freedom of </span>
<span id="cb28-665"><a href="#cb28-665" aria-hidden="true" tabindex="-1"></a><span class="in">`r broom::tidy(anova(mod))$df[1]`</span> </span>
<span id="cb28-666"><a href="#cb28-666" aria-hidden="true" tabindex="-1"></a>and</span>
<span id="cb28-667"><a href="#cb28-667" aria-hidden="true" tabindex="-1"></a><span class="in">`r broom::tidy(anova(mod))$df[3]`</span>.</span>
<span id="cb28-668"><a href="#cb28-668" aria-hidden="true" tabindex="-1"></a>This would be written as $F = $</span>
<span id="cb28-669"><a href="#cb28-669" aria-hidden="true" tabindex="-1"></a><span class="in">`r round(anova(mod)$F[1],2)`</span> ; </span>
<span id="cb28-670"><a href="#cb28-670" aria-hidden="true" tabindex="-1"></a>$d.f.= $</span>
<span id="cb28-671"><a href="#cb28-671" aria-hidden="true" tabindex="-1"></a><span class="in">`r broom::tidy(anova(mod))$df[1]`</span>, <span class="in">`r broom::tidy(anova(mod))$df[3]`</span>;</span>
<span id="cb28-672"><a href="#cb28-672" aria-hidden="true" tabindex="-1"></a>$p = $</span>
<span id="cb28-673"><a href="#cb28-673" aria-hidden="true" tabindex="-1"></a><span class="in">`r round(broom::tidy(anova(mod))$p.value[1], 3)`</span> .</span>
<span id="cb28-674"><a href="#cb28-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-675"><a href="#cb28-675" aria-hidden="true" tabindex="-1"></a><span class="fu">### Check assumptions</span></span>
<span id="cb28-676"><a href="#cb28-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-677"><a href="#cb28-677" aria-hidden="true" tabindex="-1"></a>We might expect marks to be normally distributed. However, this is a</span>
<span id="cb28-678"><a href="#cb28-678" aria-hidden="true" tabindex="-1"></a>very small sample, and choosing a non-parametric test instead would be</span>
<span id="cb28-679"><a href="#cb28-679" aria-hidden="true" tabindex="-1"></a>reasonable. However, we will continue with this example to demonstrate</span>
<span id="cb28-680"><a href="#cb28-680" aria-hidden="true" tabindex="-1"></a>how to interpret and report on the result of a parametric paired-samples</span>
<span id="cb28-681"><a href="#cb28-681" aria-hidden="true" tabindex="-1"></a>test (paired-samples *t*-test).</span>
<span id="cb28-682"><a href="#cb28-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-683"><a href="#cb28-683" aria-hidden="true" tabindex="-1"></a>A plot the residuals against the fitted values (<span class="in">`plot(mod, which = 1)`</span>)</span>
<span id="cb28-684"><a href="#cb28-684" aria-hidden="true" tabindex="-1"></a>is not useful for a paired test. The normality of the residuals should</span>
<span id="cb28-685"><a href="#cb28-685" aria-hidden="true" tabindex="-1"></a>be checked.</span>
<span id="cb28-686"><a href="#cb28-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-689"><a href="#cb28-689" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-690"><a href="#cb28-690" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> mod<span class="sc">$</span>residuals)) <span class="sc">+</span></span>
<span id="cb28-691"><a href="#cb28-691" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">3</span>)</span>
<span id="cb28-692"><a href="#cb28-692" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-693"><a href="#cb28-693" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-694"><a href="#cb28-694" aria-hidden="true" tabindex="-1"></a>We only have 10 values, so the distribution is never going to look</span>
<span id="cb28-695"><a href="#cb28-695" aria-hidden="true" tabindex="-1"></a>smooth. We can't draw strong conclusions from this, but we do at least</span>
<span id="cb28-696"><a href="#cb28-696" aria-hidden="true" tabindex="-1"></a>have a peak at 0. Similarly, a normality test is likely to be</span>
<span id="cb28-697"><a href="#cb28-697" aria-hidden="true" tabindex="-1"></a>non-significant because of the small sample size, meaning the test is</span>
<span id="cb28-698"><a href="#cb28-698" aria-hidden="true" tabindex="-1"></a>not very powerful. This means a non-significant result is not strong</span>
<span id="cb28-699"><a href="#cb28-699" aria-hidden="true" tabindex="-1"></a>evidence of the residuals following a normal distribution:</span>
<span id="cb28-700"><a href="#cb28-700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-703"><a href="#cb28-703" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-704"><a href="#cb28-704" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(mod<span class="sc">$</span>residuals)</span>
<span id="cb28-705"><a href="#cb28-705" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-706"><a href="#cb28-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-707"><a href="#cb28-707" aria-hidden="true" tabindex="-1"></a><span class="fu">### Report</span></span>
<span id="cb28-708"><a href="#cb28-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-709"><a href="#cb28-709" aria-hidden="true" tabindex="-1"></a>Individual students score significantly higher in Data Analysis than in</span>
<span id="cb28-710"><a href="#cb28-710" aria-hidden="true" tabindex="-1"></a>Biology (*t* = 2.47; *d.f.* = 9; *p* = 0.0355) with an average</span>
<span id="cb28-711"><a href="#cb28-711" aria-hidden="true" tabindex="-1"></a>difference of 7%. See @fig-marks</span>
<span id="cb28-712"><a href="#cb28-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-713"><a href="#cb28-713" aria-hidden="true" tabindex="-1"></a>::: {#fig-marks}</span>
<span id="cb28-716"><a href="#cb28-716" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-717"><a href="#cb28-717" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb28-718"><a href="#cb28-718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-719"><a href="#cb28-719" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> marks, <span class="fu">aes</span>(<span class="at">x =</span> subject, <span class="at">y =</span> mark)) <span class="sc">+</span></span>
<span id="cb28-720"><a href="#cb28-720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">pch =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb28-721"><a href="#cb28-721" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> student), <span class="at">linetype =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb28-722"><a href="#cb28-722" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> marks_summary,</span>
<span id="cb28-723"><a href="#cb28-723" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">x =</span> subject, <span class="at">y =</span> mean),</span>
<span id="cb28-724"><a href="#cb28-724" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb28-725"><a href="#cb28-725" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb28-726"><a href="#cb28-726" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Mark"</span>,</span>
<span id="cb28-727"><a href="#cb28-727" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb28-728"><a href="#cb28-728" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">110</span>)) <span class="sc">+</span></span>
<span id="cb28-729"><a href="#cb28-729" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>,</span>
<span id="cb28-730"><a href="#cb28-730" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="dv">105</span>, <span class="at">yend =</span> <span class="dv">105</span>,</span>
<span id="cb28-731"><a href="#cb28-731" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb28-732"><a href="#cb28-732" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="dv">108</span>,</span>
<span id="cb28-733"><a href="#cb28-733" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(p)<span class="sc">~</span><span class="st">"= 0.0355"</span>)) <span class="sc">+</span></span>
<span id="cb28-734"><a href="#cb28-734" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb28-735"><a href="#cb28-735" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-736"><a href="#cb28-736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-737"><a href="#cb28-737" aria-hidden="true" tabindex="-1"></a>**Students score higher in Data Analysis than in Biology**. Open circles</span>
<span id="cb28-738"><a href="#cb28-738" aria-hidden="true" tabindex="-1"></a>indicate an individual student's marks in each subject with dashed lines</span>
<span id="cb28-739"><a href="#cb28-739" aria-hidden="true" tabindex="-1"></a>joining their marks in each subject. The filled circles indicate the</span>
<span id="cb28-740"><a href="#cb28-740" aria-hidden="true" tabindex="-1"></a>mean mark for each subject. Individual students score significantly</span>
<span id="cb28-741"><a href="#cb28-741" aria-hidden="true" tabindex="-1"></a>higher in Data Analysis than in Biology (*t* = 2.47; *d.f.* = 9; *p* =</span>
<span id="cb28-742"><a href="#cb28-742" aria-hidden="true" tabindex="-1"></a>0.0355) with an average difference of 7%. Data analysis was conducted in</span>
<span id="cb28-743"><a href="#cb28-743" aria-hidden="true" tabindex="-1"></a>R <span class="co">[</span><span class="ot">@R-core</span><span class="co">]</span> with tidyverse packages <span class="co">[</span><span class="ot">@tidyverse</span><span class="co">]</span>.</span>
<span id="cb28-744"><a href="#cb28-744" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb28-745"><a href="#cb28-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-746"><a href="#cb28-746" aria-hidden="true" tabindex="-1"></a><span class="fu">## Two paired-samples, non-parametric</span></span>
<span id="cb28-747"><a href="#cb28-747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-748"><a href="#cb28-748" aria-hidden="true" tabindex="-1"></a>We have the marks for just 10 students. This sample is too small for us</span>
<span id="cb28-749"><a href="#cb28-749" aria-hidden="true" tabindex="-1"></a>to judge whether the marks are normally distributed. We will use a</span>
<span id="cb28-750"><a href="#cb28-750" aria-hidden="true" tabindex="-1"></a>non-parametric test instead. The "Wilcoxon signed-rank" test is the</span>
<span id="cb28-751"><a href="#cb28-751" aria-hidden="true" tabindex="-1"></a>non-parametric equivalent of the paired-samples *t*-test. This is often</span>
<span id="cb28-752"><a href="#cb28-752" aria-hidden="true" tabindex="-1"></a>referred to as the paired-sample Wilcoxon test, or just the Wilcoxon</span>
<span id="cb28-753"><a href="#cb28-753" aria-hidden="true" tabindex="-1"></a>test.</span>
<span id="cb28-754"><a href="#cb28-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-755"><a href="#cb28-755" aria-hidden="true" tabindex="-1"></a>The test is also applied in R with the <span class="in">`wilcox.test()`</span> function but we</span>
<span id="cb28-756"><a href="#cb28-756" aria-hidden="true" tabindex="-1"></a>add the <span class="in">`paired = TRUE`</span> argument. We also have to give the two datasets</span>
<span id="cb28-757"><a href="#cb28-757" aria-hidden="true" tabindex="-1"></a>rather than using the "formula method" of <span class="in">`mark ~ subject`</span>. This means</span>
<span id="cb28-758"><a href="#cb28-758" aria-hidden="true" tabindex="-1"></a>it is useful to pivot the data to "wide" format.</span>
<span id="cb28-759"><a href="#cb28-759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-760"><a href="#cb28-760" aria-hidden="true" tabindex="-1"></a><span class="fu">### Pivot wider</span></span>
<span id="cb28-761"><a href="#cb28-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-762"><a href="#cb28-762" aria-hidden="true" tabindex="-1"></a>Create a new dataframe marks_wide from marks:</span>
<span id="cb28-763"><a href="#cb28-763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-766"><a href="#cb28-766" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-767"><a href="#cb28-767" aria-hidden="true" tabindex="-1"></a>marks_wide <span class="ot">&lt;-</span> marks <span class="sc">|&gt;</span> </span>
<span id="cb28-768"><a href="#cb28-768" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">values_from =</span> mark, </span>
<span id="cb28-769"><a href="#cb28-769" aria-hidden="true" tabindex="-1"></a>              <span class="at">names_from =</span> subject, </span>
<span id="cb28-770"><a href="#cb28-770" aria-hidden="true" tabindex="-1"></a>              <span class="at">id_cols =</span> student)</span>
<span id="cb28-771"><a href="#cb28-771" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-772"><a href="#cb28-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-773"><a href="#cb28-773" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`values_from = mark`</span>: Uses the <span class="in">`mark`</span> column as values in the wide</span>
<span id="cb28-774"><a href="#cb28-774" aria-hidden="true" tabindex="-1"></a>    format.</span>
<span id="cb28-775"><a href="#cb28-775" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`names_from = subject`</span>: Creates new columns based on unique values</span>
<span id="cb28-776"><a href="#cb28-776" aria-hidden="true" tabindex="-1"></a>    in <span class="in">`subject`</span></span>
<span id="cb28-777"><a href="#cb28-777" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`id_cols = student`</span>: Keeps <span class="in">`student`</span> as the identifier (*i.e.*, each</span>
<span id="cb28-778"><a href="#cb28-778" aria-hidden="true" tabindex="-1"></a>    row represents a student).</span>
<span id="cb28-779"><a href="#cb28-779" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-780"><a href="#cb28-780" aria-hidden="true" tabindex="-1"></a><span class="fu">### Apply `wilcox.test()`</span></span>
<span id="cb28-781"><a href="#cb28-781" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-782"><a href="#cb28-782" aria-hidden="true" tabindex="-1"></a>To apply a paired test with <span class="in">`wilcox.test()`</span> we need to use the wide</span>
<span id="cb28-783"><a href="#cb28-783" aria-hidden="true" tabindex="-1"></a>format data (untidy) and add the <span class="in">`paired = TRUE`</span> argument:</span>
<span id="cb28-784"><a href="#cb28-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-787"><a href="#cb28-787" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-788"><a href="#cb28-788" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(marks_wide<span class="sc">$</span>Biology, marks_wide<span class="sc">$</span>DataAnalysis, <span class="at">paired =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-789"><a href="#cb28-789" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-790"><a href="#cb28-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-791"><a href="#cb28-791" aria-hidden="true" tabindex="-1"></a><span class="fu">### Report</span></span>
<span id="cb28-792"><a href="#cb28-792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-793"><a href="#cb28-793" aria-hidden="true" tabindex="-1"></a>Individual students score significantly higher in Data Analysis than in</span>
<span id="cb28-794"><a href="#cb28-794" aria-hidden="true" tabindex="-1"></a>Biology (Wilcoxon signed rank test: *V* = 6.5; $n$ = 10; *p* = 0.036).</span>
<span id="cb28-795"><a href="#cb28-795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-796"><a href="#cb28-796" aria-hidden="true" tabindex="-1"></a><span class="fu">## Summary</span></span>
<span id="cb28-797"><a href="#cb28-797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-798"><a href="#cb28-798" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>A linear model with one explanatory variable with two groups and one</span>
<span id="cb28-799"><a href="#cb28-799" aria-hidden="true" tabindex="-1"></a>    continuous response is "a two-sample test".</span>
<span id="cb28-800"><a href="#cb28-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-801"><a href="#cb28-801" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>If pairs of observations in the groups have something in common that</span>
<span id="cb28-802"><a href="#cb28-802" aria-hidden="true" tabindex="-1"></a>    make them more similar to each other, than to other observations,</span>
<span id="cb28-803"><a href="#cb28-803" aria-hidden="true" tabindex="-1"></a>    then those observations are not independent. A **paired-samples</span>
<span id="cb28-804"><a href="#cb28-804" aria-hidden="true" tabindex="-1"></a>    test** is used when the observations are not independent.</span>
<span id="cb28-805"><a href="#cb28-805" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-806"><a href="#cb28-806" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>A linear model with one explanatory variable with two groups and one</span>
<span id="cb28-807"><a href="#cb28-807" aria-hidden="true" tabindex="-1"></a>    continuous response is also known as a **two-sample *t*-test** when</span>
<span id="cb28-808"><a href="#cb28-808" aria-hidden="true" tabindex="-1"></a>    the samples are independent and as a **paired-samples *t*-test**</span>
<span id="cb28-809"><a href="#cb28-809" aria-hidden="true" tabindex="-1"></a>    when they are not</span>
<span id="cb28-810"><a href="#cb28-810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-811"><a href="#cb28-811" aria-hidden="true" tabindex="-1"></a><span class="ss">4.  </span>We can use <span class="in">`lm()`</span> to do two-sample and paired sample tests. We can</span>
<span id="cb28-812"><a href="#cb28-812" aria-hidden="true" tabindex="-1"></a>    also use <span class="in">`t.test()`</span> for these but using <span class="in">`lm()`</span> helps us understand</span>
<span id="cb28-813"><a href="#cb28-813" aria-hidden="true" tabindex="-1"></a>    tests with more groups and/or more variables where we will have to</span>
<span id="cb28-814"><a href="#cb28-814" aria-hidden="true" tabindex="-1"></a>    use <span class="in">`lm()`</span>. The output of <span class="in">`lm()`</span> is also more typical of the output</span>
<span id="cb28-815"><a href="#cb28-815" aria-hidden="true" tabindex="-1"></a>    of statistical functions in R.</span>
<span id="cb28-816"><a href="#cb28-816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-817"><a href="#cb28-817" aria-hidden="true" tabindex="-1"></a><span class="ss">5.  </span>We estimate the **coefficients** (also called the **parameters**) of</span>
<span id="cb28-818"><a href="#cb28-818" aria-hidden="true" tabindex="-1"></a>    the model. For a two-sample test these are the mean of the first</span>
<span id="cb28-819"><a href="#cb28-819" aria-hidden="true" tabindex="-1"></a>    group, $\beta_0$ (which might also be called the intercept) and the</span>
<span id="cb28-820"><a href="#cb28-820" aria-hidden="true" tabindex="-1"></a>    difference between the means of the first and second groups,</span>
<span id="cb28-821"><a href="#cb28-821" aria-hidden="true" tabindex="-1"></a>    $\beta_1$ (which might also be called the slope). For a</span>
<span id="cb28-822"><a href="#cb28-822" aria-hidden="true" tabindex="-1"></a>    paired-sample test there is just one parameter, the mean difference</span>
<span id="cb28-823"><a href="#cb28-823" aria-hidden="true" tabindex="-1"></a>    between pairs of values, $\beta_0$ (which might also be called the</span>
<span id="cb28-824"><a href="#cb28-824" aria-hidden="true" tabindex="-1"></a>    intercept). We test whether the parameters differ significantly from</span>
<span id="cb28-825"><a href="#cb28-825" aria-hidden="true" tabindex="-1"></a>    zero</span>
<span id="cb28-826"><a href="#cb28-826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-827"><a href="#cb28-827" aria-hidden="true" tabindex="-1"></a><span class="ss">6.  </span>We can use <span class="in">`lm()`</span> to a linear regression.</span>
<span id="cb28-828"><a href="#cb28-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-829"><a href="#cb28-829" aria-hidden="true" tabindex="-1"></a><span class="ss">7.  </span>In the output of <span class="in">`lm()`</span> the coefficients are listed in a table in</span>
<span id="cb28-830"><a href="#cb28-830" aria-hidden="true" tabindex="-1"></a>    the Estimates column. The *p*-value for each coefficient is in the</span>
<span id="cb28-831"><a href="#cb28-831" aria-hidden="true" tabindex="-1"></a>    test of whether it differs from zero. At the bottom of the output</span>
<span id="cb28-832"><a href="#cb28-832" aria-hidden="true" tabindex="-1"></a>    there is a test of the model *overall*. In this case, this is</span>
<span id="cb28-833"><a href="#cb28-833" aria-hidden="true" tabindex="-1"></a>    exactly the same as the test of the $\beta_1$ and the p-values are</span>
<span id="cb28-834"><a href="#cb28-834" aria-hidden="true" tabindex="-1"></a>    identical. The R-squared value is the proportion of the variance in</span>
<span id="cb28-835"><a href="#cb28-835" aria-hidden="true" tabindex="-1"></a>    the response variable that is explained by the model.</span>
<span id="cb28-836"><a href="#cb28-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-837"><a href="#cb28-837" aria-hidden="true" tabindex="-1"></a><span class="ss">8.  </span>The assumptions of the general linear model are that the residuals</span>
<span id="cb28-838"><a href="#cb28-838" aria-hidden="true" tabindex="-1"></a>    are normally distributed and have homogeneity of variance. A</span>
<span id="cb28-839"><a href="#cb28-839" aria-hidden="true" tabindex="-1"></a>    residual is the difference between the predicted value and the</span>
<span id="cb28-840"><a href="#cb28-840" aria-hidden="true" tabindex="-1"></a>    observed value.</span>
<span id="cb28-841"><a href="#cb28-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-842"><a href="#cb28-842" aria-hidden="true" tabindex="-1"></a><span class="ss">9.  </span>We examine a histogram of the residuals and use the Shapiro-Wilk</span>
<span id="cb28-843"><a href="#cb28-843" aria-hidden="true" tabindex="-1"></a>    normality test to check the normality assumption. We check the</span>
<span id="cb28-844"><a href="#cb28-844" aria-hidden="true" tabindex="-1"></a>    variance of the residuals is the same for all fitted values with a</span>
<span id="cb28-845"><a href="#cb28-845" aria-hidden="true" tabindex="-1"></a>    residuals vs fitted plot.</span>
<span id="cb28-846"><a href="#cb28-846" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-847"><a href="#cb28-847" aria-hidden="true" tabindex="-1"></a><span class="ss">10. </span>If the assumptions are not met, we can use alternatives known as</span>
<span id="cb28-848"><a href="#cb28-848" aria-hidden="true" tabindex="-1"></a>    non-parametric tests. These are applied with <span class="in">`wilcox.test()`</span> in R.</span>
<span id="cb28-849"><a href="#cb28-849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-850"><a href="#cb28-850" aria-hidden="true" tabindex="-1"></a><span class="ss">11. </span>When reporting the results of a test we give the significance,</span>
<span id="cb28-851"><a href="#cb28-851" aria-hidden="true" tabindex="-1"></a>    direction and size of the effect. Our figures and the values we give</span>
<span id="cb28-852"><a href="#cb28-852" aria-hidden="true" tabindex="-1"></a>    should reflect the type of test we have used. We use means and</span>
<span id="cb28-853"><a href="#cb28-853" aria-hidden="true" tabindex="-1"></a>    standard errors for parametric tests and medians and interquartile</span>
<span id="cb28-854"><a href="#cb28-854" aria-hidden="true" tabindex="-1"></a>    ranges for non-parametric tests. We also give the test statistic,</span>
<span id="cb28-855"><a href="#cb28-855" aria-hidden="true" tabindex="-1"></a>    the degrees of freedom (parametric) or sample size (non-parametric)</span>
<span id="cb28-856"><a href="#cb28-856" aria-hidden="true" tabindex="-1"></a>    and the p-value. We annotate our figures with the p-value, making</span>
<span id="cb28-857"><a href="#cb28-857" aria-hidden="true" tabindex="-1"></a>    clear which comparison it applies to.</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Rand, E. (2025). Computational Analysis for Bioscientists<br>
(Version 0.2) https://3mmarand.github.io/comp4biosci/</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/two-sample-tests.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer><script>
document.addEventListener("DOMContentLoaded", function() {
  // grab all links under “Other Links” in the right-hand TOC
  const links = document.querySelectorAll('.quarto-other-links ul li a');
  links.forEach(function(a) {
    // find the one whose text is exactly “Make a suggestion”
    if (a.textContent.trim() === 'Make a suggestion') {
      const pageUrl   = encodeURIComponent(window.location.href);
      const pageTitle = encodeURIComponent(document.title);
      a.href = 'https://github.com/3mmaRand/comp4biosci/issues/new'
             + '?title=' + pageTitle
             + '&body='  + pageUrl;
      a.target = '_blank';
    }
  });
});
</script><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


</body></html>